<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Expressive Interfaces for Multi-omics Simulation</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs-2.25/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="css/xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


class: title
background-size: cover




&lt;div id="links"&gt;
Slides: &lt;a href="https://go.wisc.edu/4y9176"&gt;go.wisc.edu/4y9176&lt;/a&gt;
&lt;/div&gt;
&lt;div id="title"&gt;
Expressive Interfaces for Multi-omics Simulation
&lt;/div&gt;

&lt;br/&gt;

&lt;i&gt;As the program notes say, "Dive into the gene pool to see evolution in action," and enjoy "genetic engineering in the privacy of your own home." 

Requires: 2.5 megabytes of system memory or 3 megabytes under System 7, and a hard drive.&lt;/i&gt;

-- The New York Times, 1992, reviewing the newly released SimLife

&lt;br/&gt;

&lt;i&gt;Heard melodies are sweet, but those unheard &lt;br/&gt;
Are sweeter; therefore, ye soft pipes, play on... &lt;br/&gt;&lt;/i&gt;
-- Keats

&lt;div id="subtitle"&gt;
Kris Sankaran &lt;br/&gt;
&lt;a href="https://go.wisc.edu/pgb8nl"&gt;go.wisc.edu/pgb8nl&lt;/a&gt; &lt;br/&gt;
06 | December | 2023 &lt;br/&gt;
UW-Madison Statistics Seminar &lt;br/&gt;
&lt;/div&gt;

---

### Why Simulate?

* **Experimental Design**: We have to decide on cohorts, longitudinal sampling plans, and sequencing technologies, not to mention sample sizes.

* **Benchmarking**: They allow us to benchmark algorithms even when ground truth labels are unavailable.

* **Data Augmentation**: Simulated samples can improve algorithmic performance and integration.

* **Calibration**: Statistical performance on simulated data can help us calibrate workflows to improve power and control the false discovery rate.

---

### Why Simulate?

* **Experimental Design**: We have to decide on cohorts, longitudinal sampling plans, and sequencing technologies, not to mention sample sizes.

* **Benchmarking**: They allow us to benchmark algorithms even when ground truth labels are unavailable.

* **Data Augmentation**: Simulated samples can improve algorithmic performance and integration.

* **Calibration**: Statistical performance on simulated data can help us calibrate workflows to improve power and control the false discovery rate.

&lt;g style="font-size: 20px; margin: 0; line-height: 30px; display: block;"&gt;
Examples: BASICS,
compcodeR,
deconvR,
dropsim,
ESCO,
FreeHi-C,
FreeHiCLite,
hierarchicell,
kersplat,
metaART,
MOSim,
MSstatsSampleSize,
MIDAS,
Mimesys,
multiomics_networks_simulation,
muscat,
powsimR,
POWSC,
SCDD,
scDesign3,
SCRIP,
Sim3C,
SimATAC,
SimFFPE,
sismonr,
spaSim,
sparseDOSSA,
Splat,
SPARSim,
SPsimSeq,
SparseDC,
SymSim,
ZINB-WaVE,
zingeR, ...
&lt;/g&gt;

---

### A Grand Challenge

*Many single-cell data analysis packages include their own ad hoc data simulators [111, 211, 241, 264, 349–353]. However, these simulators are usually not available as separate tools or even as a source code, tailored to specific problems studied in corresponding papers and sometimes not comprehensively documented, thus limiting their utility for the broad research community.*

-- Challenge 11 from "Eleven grand challenges in single-cell data science" [1].

---

### That's a Bit Uncharitable

1. It's true that most researchers do not routinely use simulation. The issue
isn't sloppiness in sharing or documenting code, though.

1. The deeper issue is that we usually view simulators as monolithic. We can
adjust parameters but can't interchange components. This makes them hard to
adapt to new technologies or applications.

1. I will show two examples of well-designed simulators. How would you go about
trying to extend them?
    
---

### Existing Interfaces

This is from the [`splatter` introductory vignette](https://bioconductor.org/packages/devel/bioc/vignettes/splatter/inst/doc/splatter.html#2_Quickstart) [2]. 
Of the packages I've listed, it has most thoughtful interface.


```r
params &lt;- setParams(params, mean.shape = 0.5, de.prob = 0.2)
params
#&gt; A Params object of class SplatParams 
#&gt; Parameters can be (estimable) or [not estimable], 'Default' or  'NOT DEFAULT' 
#&gt; Secondary parameters are usually set during simulation
#&gt; 
#&gt; Global: 
#&gt; (GENES)  (Cells)   [SEED] 
#&gt;    8000      100   694289 
#&gt; 
#&gt; 29 additional parameters 
#&gt; 
#&gt; Batches: 
#&gt;     [Batches]  [Batch Cells]     [Location]        [Scale]       [Remove] 
#&gt;             1            100            0.1            0.1          FALSE 
#&gt; 
#&gt; Mean: 
#&gt;  (RATE)  (SHAPE) 
#&gt;     0.5      0.5 
```

---

### Existing Interfaces

.pull-left[
Here is an example from the [`scDesign3` Quick Start](https://songdongyuan1994.github.io/scDesign3/docs/articles/scDesign3.html) [3].
This package has the most generally applicable statistical methodology.
]

.pull-right[

```r
example_simu &lt;- scdesign3(
    sce = example_sce,
    assay_use = "counts",
    celltype = "cell_type",
    pseudotime = "pseudotime",
    spatial = NULL,
    other_covariates = NULL,
    mu_formula = "s(pseudotime, k = 10, bs = 'cr')",
    sigma_formula = "1", # If you want your dispersion also varies along pseudotime, use "s(pseudotime, k = 5, bs = 'cr')"
    family_use = "nb",
    n_cores = 2,
    usebam = FALSE,
    corr_formula = "1",
    copula = "gaussian",
    DT = TRUE,
    pseudo_obs = FALSE,
    return_model = FALSE,
    nonzerovar = FALSE
  )
```
]

---

&lt;g style="font-size: 36px"&gt;
&lt;b&gt;Main Idea: Apply interactive computing principles to multi-omics simulation.&lt;/b&gt;
&lt;/g&gt;

&lt;img src="figures/Lego_Brick_4.svg" width=50/&gt; &lt;span style="color:#025E73"&gt;Modularity&lt;/span&gt;: We should be able to build
problem-specific simulators by composing simple pieces.

&lt;img src="figures/computer_mouse.png" width=35/&gt; &lt;span style="color:#D94E4E"&gt;Interactivity&lt;/span&gt;: We should give domain
researchers agency in designing, evaluating, and modifying statistical
hypotheses.

&lt;img src="figures/speech_bubbles.png" width=70/&gt; &lt;span style="color:#378C5C"&gt;Communication&lt;/span&gt;: Model-building is a social
activity, and well-designed tools encourage substantive community engagement.

---

### `scDesign3` Review

The rest of the talk will be about using these principles to develop a new
version of `scDesign3`. Let's review that package's approach.

.center[
&lt;img src="figures/scdesignOverview.png" width=800/&gt;
]

First, we estimate models `\(\hat{F}_{g}\left(y_{i} \vert \mathbf{x}_{i}\right)\)` for each gene `\(g\)`.

* Can use a variety of families: Gaussian, Poisson, Negative Binomial,...
* Can learn relationships for each parameter `\(\theta\left(\mathbf{x}_{i}\right)\)`.

---

### `scDesign3` Review

.pull-left[
&lt;img src="figures/scdesignOverview2.png" width=400/&gt;
]

.pull-right[
Model the joint distribution of quantiles using a copula model.
* The existing package supports Gaussian and Vine copulas.
* Can estimate separate copulas across groups to capture differential dependence.
]

---

.section[
## Interface Design
]

---

### Nouns &amp; Verbs

We can break the interface design question into two parts.

1. **Data Structures**: How can we represent the simulator in a way that is both
transparent to a human and precise enough for a computer?

1. **Operations**: How can we encourage users to study and tinker with the data
structures?

If the resulting grammar is expressive enough, then researchers will be able to
solve problems we may not have anticipated.



---

### Nouns: Gene-Level Models

For each gene, we can specify the regression formula and distributional family.


```r
margins &lt;- setup_margins(sce, ~ ns(pseudotime, 3),  ~ ZINBLSS())
margins
```

&lt;pre class="r-output"&gt;&lt;code&gt;## &lt;span style='color: #BB00BB;'&gt;Plan:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 6 x 3&lt;/span&gt;
##     feature              family               link
##   &lt;span style='color: #555555; font-style: italic;'&gt;&lt;gene_id&gt;&lt;/span&gt;             &lt;span style='color: #555555; font-style: italic;'&gt;&lt;distn&gt;&lt;/span&gt;             &lt;span style='color: #555555; font-style: italic;'&gt;&lt;link&gt;&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;1&lt;/span&gt;       &lt;span style='color: #00BBBB;'&gt;Pyy&lt;/span&gt; ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;2&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Iapp&lt;/span&gt; ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;3&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Chgb&lt;/span&gt; ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;4&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Rbp4&lt;/span&gt; ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;5&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Spp1&lt;/span&gt; ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;6&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Chga&lt;/span&gt; ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #00BBBB;'&gt;Pyy, Iapp, Chgb&lt;/span&gt;, and &lt;span style='color: #00BBBB;'&gt;7&lt;/span&gt; &lt;span style='color: #00BBBB;'&gt;other features&lt;/span&gt; need fitting.&lt;span style='color: #BB00BB;'&gt;
## Estimates:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 0 x 0&lt;/span&gt;
&lt;/code&gt;&lt;/pre&gt;



---

### Nouns: Copula Models

We can tie together a collection of marginals using a copula model.


```r
simulator &lt;- new(
  "JointDistribution", 
  margins = margins, 
  dependence = copula_gaussian()
)
```

We can support different correlation structures across groups.


```r
simulator &lt;- new(
  "JointDistribution", 
  margins = margins, 
  dependence = copula_gaussian(~ cell_type)
)
```



---

### Extensibility

For the marginals, we can borrow from existing LSS packages (gamlss [4], gamboostLSS [5]) to implement gene-wise regressions,

* Gaussian, Gamma, Poisson, Binomial, Negative-Binomial, Zero-Inflated Poisson, Zero Inflated Negative Binomial,…

We can also build a unified interface to various copula estimation routines,

* Vine copulas for capturing higher-order moments.
* Gaussian and Student-t copulas with standard sample covariance, shrunken covariance, or graphical lasso precision estimates.
* `copula_vine(), copula_adaptive(), copula_glasso(), ...`

---

### Verbs: &lt;span style="color: #D94E4E"&gt;Plot&lt;/span&gt;



```r
plot(simulator, "pseudotime", line_opts = line_opts, ribbon_opts = ribbon_opts, point_opts = point_opts) +
  facet_wrap(~ feature, ncol = 4)
```

&lt;img src="20231206_files/figure-html/example_plot-1.png" width="800" style="display: block; margin: auto;" /&gt;

---

### Verbs: &lt;span style="color:#025E73"&gt;Mutate&lt;/span&gt;

`mutate` lets you modify a few elements from a larger simulator.


```r
altered &lt;- mutate(margins, Chga:Ins1, link = ~ pseudotime) |&gt;
  estimate(sce)
```
&lt;img src="20231206_files/figure-html/plot_alteration-1.png" width="800" style="display: block; margin: auto;" /&gt;

---

### Verbs: &lt;span style="color:#025E73"&gt;Mutate&lt;/span&gt;

We can layer several changes on top of one another. Syntax color highlights
modifications that haven’t been re-fit.



```r
margins |&gt;
  mutate(Chga:Ins1, link = ~ pseudotime) |&gt;
  mutate(c("Pyy", "Rbp4"), family = ~ GaussianLSS()) |&gt;
  mutate(Iapp, link = list(mu = ~ ns(pseudotime, df = 5), sigma = ~ 1, nu = ~ 1))
```

&lt;pre class="r-output"&gt;&lt;code&gt;## &lt;span style='color: #BB00BB;'&gt;Plan:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 6 x 3&lt;/span&gt;
##     feature              family               link
##   &lt;span style='color: #555555; font-style: italic;'&gt;&lt;gene_id&gt;&lt;/span&gt;             &lt;span style='color: #555555; font-style: italic;'&gt;&lt;distn&gt;&lt;/span&gt;             &lt;span style='color: #555555; font-style: italic;'&gt;&lt;link&gt;&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;1&lt;/span&gt;       &lt;span style='color: #00BBBB;'&gt;Pyy&lt;/span&gt; Gaussian [mu,sigma] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;2&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Iapp&lt;/span&gt; ZINBI [mu,sigma,nu]  mu~ns(pseudoti...
## &lt;span style='color: #555555;'&gt;3&lt;/span&gt;      Chgb ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;4&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Rbp4&lt;/span&gt; Gaussian [mu,sigma] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;5&lt;/span&gt;      Spp1 ZINBI [mu,sigma,nu] ~ns(pseudotime, 3)
## &lt;span style='color: #555555;'&gt;6&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;Chga&lt;/span&gt; ZINBI [mu,sigma,nu]        ~pseudotime
## &lt;span style='color: #00BBBB;'&gt;Pyy, Iapp, Rbp4&lt;/span&gt;, and &lt;span style='color: #00BBBB;'&gt;3&lt;/span&gt; &lt;span style='color: #00BBBB;'&gt;other features&lt;/span&gt; need fitting.&lt;span style='color: #BB00BB;'&gt;
## Estimates:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 3 x 2&lt;/span&gt;
##   feature fit       
##   &lt;span style='color: #555555; font-style: italic;'&gt;&lt;chr&gt;&lt;/span&gt;   &lt;span style='color: #555555; font-style: italic;'&gt;&lt;list&gt;&lt;/span&gt;    
## &lt;span style='color: #555555;'&gt;1&lt;/span&gt; Pyy     &lt;span style='color: #555555;'&gt;&lt;glmbsLSS&gt;&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;2&lt;/span&gt; Iapp    &lt;span style='color: #555555;'&gt;&lt;glmbsLSS&gt;&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;3&lt;/span&gt; Chgb    &lt;span style='color: #555555;'&gt;&lt;glmbsLSS&gt;&lt;/span&gt;
## ... and 7 additional features.
&lt;/code&gt;&lt;/pre&gt;

---

### Verbs: &lt;span style="color:#025E73"&gt;Mutate&lt;/span&gt;

1. Here is a more realistic example from a longitudinal microbiome study.
2. We can use `mutate` to define a synthetic null with no disease effect for a known subset of genes.

.pull-three-quarters-left[
&lt;img src="figures/nulls_unaltered.png"/&gt;
]
.pull-three-quarters-right[
&lt;img src="figures/pairwise_cors.png"/&gt;
]

---

### Verbs: &lt;span style="color:#025E73"&gt;Mutate&lt;/span&gt;

1. Here is a more realistic example from a longitudinal microbiome study.
2. We can use `mutate` to define a synthetic null with no disease effect for a known subset of genes.

.pull-three-quarters-left[
&lt;img src="figures/altered_ns.png"/&gt;
]
.pull-three-quarters-right[
&lt;img src="figures/pairwise_cors_altered.png"/&gt;
]

---

### Verbs: &lt;span style="color:#025E73"&gt;Join&lt;/span&gt;

We should make it possible to combine simulators like Lego blocks.



```r
experiments &lt;- list(methylation = SCGEMMETH_sce, rna = SCGEMRNA_sce)
families &lt;- list(~ BI(), ~ GaussianLSS())
sims &lt;- experiments |&gt;
  map2(families, \(x, y) setup_simulator(x, ~ cell_type, y))
```

&lt;img src="figures/simulator_join_motivation.png"/&gt;

---

### Verbs: &lt;span style="color:#025E73"&gt;Join&lt;/span&gt; (Copula)

One approach is to merge the list of marginal distributions and re-estimate the joint distribution.


```r
sim_joined &lt;- map(sims, estimate, nu = 0.1) |&gt;
  join_copula(copula_glasso())
```

This assumes that we have samples where all features are measured.

.center[
&lt;img src="figures/copula_join.png" width = 680/&gt;
]

---

### Verbs: &lt;span style="color:#025E73"&gt;Join&lt;/span&gt; (Conditioning)

Alternatively, we can combine two simulators by conditioning them on shared latent structure.


```r
sim_joined &lt;- join_pamona(sims)
```

&lt;pre class="r-output"&gt;&lt;code&gt;## Pamona start!
## use random seed: 1
## Pamona Done! takes 3.288163 seconds
&lt;/code&gt;&lt;/pre&gt;

.center[
&lt;img src="figures/simulator_join.png" width = 800/&gt;
]

---

### Verbs: &lt;span style="color:#025E73"&gt;Join&lt;/span&gt; (Conditioning)

This used partial manifold alignment [6] to
learn shared latent variables across assays and works even in the diagonal
integration setting.


```r
sim_joined
```

&lt;pre class="r-output"&gt;&lt;code&gt;## $methylation
## &lt;span style='color: #555555;'&gt;[Marginals]
## &lt;/span&gt;&lt;span style='color: #BB00BB;'&gt;Plan:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 6 x 3&lt;/span&gt;
##           feature  family                       link
##         &lt;span style='color: #555555; font-style: italic;'&gt;&lt;gene_id&gt;&lt;/span&gt; &lt;span style='color: #555555; font-style: italic;'&gt;&lt;distn&gt;&lt;/span&gt;                     &lt;span style='color: #555555; font-style: italic;'&gt;&lt;link&gt;&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;1&lt;/span&gt;        &lt;span style='color: #00BBBB;'&gt;AK123759&lt;/span&gt; BI [mu] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;2&lt;/span&gt;          &lt;span style='color: #00BBBB;'&gt;ADAM33&lt;/span&gt; BI [mu] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;3&lt;/span&gt;            &lt;span style='color: #00BBBB;'&gt;NFIX&lt;/span&gt; BI [mu] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;4&lt;/span&gt;           &lt;span style='color: #00BBBB;'&gt;FOXD2&lt;/span&gt; BI [mu] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;5&lt;/span&gt;  &lt;span style='color: #00BBBB;'&gt;HLX.U.2212.AS1&lt;/span&gt; BI [mu] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;6&lt;/span&gt; &lt;span style='color: #00BBBB;'&gt;LY86.U.2212.AS1&lt;/span&gt; BI [mu] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #00BBBB;'&gt;AK123759, ADAM33, NFIX&lt;/span&gt;, and &lt;span style='color: #00BBBB;'&gt;24&lt;/span&gt; &lt;span style='color: #00BBBB;'&gt;other features&lt;/span&gt; need fitting.&lt;span style='color: #BB00BB;'&gt;
## Estimates:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 0 x 0&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;
## [Dependence]
## 
## [Template Data]
## &lt;/span&gt;class: SingleCellExperiment 
## dim: 27 142 
## metadata(0):
## assays(1): counts
## rownames(27): AK123759 ADAM33 ... KCNQ2 CDH22
## rowData names(0):
## colnames(142): CellMeth1 CellMeth2 ... CellMeth141 CellMeth142
## colData names(10): X1 X2 ... UMAP1 UMAP2
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## 
## $rna
## &lt;span style='color: #555555;'&gt;[Marginals]
## &lt;/span&gt;&lt;span style='color: #BB00BB;'&gt;Plan:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 6 x 3&lt;/span&gt;
##     feature              family                       link
##   &lt;span style='color: #555555; font-style: italic;'&gt;&lt;gene_id&gt;&lt;/span&gt;             &lt;span style='color: #555555; font-style: italic;'&gt;&lt;distn&gt;&lt;/span&gt;                     &lt;span style='color: #555555; font-style: italic;'&gt;&lt;link&gt;&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;1&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;ZIC3&lt;/span&gt; Gaussian [mu,sigma] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;2&lt;/span&gt;     &lt;span style='color: #00BBBB;'&gt;KCNQ2&lt;/span&gt; Gaussian [mu,sigma] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;3&lt;/span&gt;     &lt;span style='color: #00BBBB;'&gt;ZFP42&lt;/span&gt; Gaussian [mu,sigma] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;4&lt;/span&gt;      &lt;span style='color: #00BBBB;'&gt;OTX2&lt;/span&gt; Gaussian [mu,sigma] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;5&lt;/span&gt;     &lt;span style='color: #00BBBB;'&gt;SALL4&lt;/span&gt; Gaussian [mu,sigma] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #555555;'&gt;6&lt;/span&gt;     &lt;span style='color: #00BBBB;'&gt;NANOG&lt;/span&gt; Gaussian [mu,sigma] ~cell_type + UMAP1 + UMAP2
## &lt;span style='color: #00BBBB;'&gt;ZIC3, KCNQ2, ZFP42&lt;/span&gt;, and &lt;span style='color: #00BBBB;'&gt;29&lt;/span&gt; &lt;span style='color: #00BBBB;'&gt;other features&lt;/span&gt; need fitting.&lt;span style='color: #BB00BB;'&gt;
## Estimates:
## &lt;/span&gt;&lt;span style='color: #555555;'&gt;# A tibble: 0 x 0&lt;/span&gt;
## &lt;span style='color: #555555;'&gt;
## [Dependence]
## 
## [Template Data]
## &lt;/span&gt;class: SingleCellExperiment 
## dim: 32 177 
## metadata(0):
## assays(1): logcounts
## rownames(32): ZIC3 KCNQ2 ... NESTIN MYC
## rowData names(0):
## colnames(177): CellRNA1 CellRNA2 ... CellRNA176 CellRNA177
## colData names(10): X1 X2 ... UMAP1 UMAP2
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
&lt;/code&gt;&lt;/pre&gt;

---

.section[
## Interactive Demo
]
&lt;br/&gt;
&lt;br/&gt;
&lt;br/&gt;

&lt;div style="font-size: 34px;"&gt;
Live Code: &lt;a href="https://go.wisc.edu/"&gt;https://go.wisc.edu/&lt;/a&gt;
&lt;br/&gt;
Colab Notebook: &lt;a href="https://go.wisc.edu/u5a94m"&gt;https://go.wisc.edu/u5a94m&lt;/a&gt;
&lt;br/&gt;
Compiled Quarto: &lt;a href="https://go.wisc.edu/62eh36"&gt; https://go.wisc.edu/62eh36&lt;/a&gt;
&lt;/div&gt;

---

### Example: Multi-Study Comparison

From [7] by Steve Goodman:

&gt; Imagine a number that does not tell us what we know, but how much we have learned… [W]e should incorporate a Bayesian framework into our writing, and not just our speaking. We should describe our data as one source of information among many that make a relationship either plausible or unlikely. The use of summaries such as the Bayes factor encourages that, while use of the P-value makes it nearly impossible. 

---

### Example: Multi-Study Comparison

1. Simulators offer a way of encoding community consensus. We are interested in how evidence accumulates and consensus evolves.
2. We will work with the `curatedMetagenomicData` [8], which includes 12 colorectal
cancer (CRC) studies processed with the same pipeline.

---

### Association from first study

For each taxon `\(j\)`, we fit two `\(\text{ZINB}\left(\mu_{j}, \sigma_{j}, \nu_{j}\right)\)` models:

`\begin{align}
H_{j0}: \left(\mu_j, \sigma_j,\nu_j\right) &amp;\sim \log\left(\text{Sequencing Depth}\right) \\
H_{j1}: \left(\mu_j, \sigma_j,\nu_j\right) &amp;\sim \log\left(\text{Sequencing Depth}\right) + \text{CRC}
\end{align}`

Given data `\(X^{(1)}\)` from study 1, we compute `\(P\left(H_{j1} \vert X^{(1)}\right)\)`, assuming `\(P\left(H_{j1}\right) = 0.1\)` a priori and that `\(H_{j1}\)` and `\(H_{j0}\)` are the only possible states of the world.

---

### Association from first study

These are the taxa with the largest value of `\(P\left(H_{j1} \vert X^{(1)}\right)\)`.

.center[
&lt;img src="figures/initial_selection.png" width=750/&gt;
]

---

### Remaining Studies

We can see how this changes as the number of studies is increased. We now consider,
`\begin{align}
H_{j0}: \left(\mu_j, \sigma_j,\nu_j\right) &amp;\sim \log\left(\text{Sequencing Depth}\right) + \text{Study ID} \\
H_{j1}: \left(\mu_j, \sigma_j,\nu_j\right) &amp;\sim \log\left(\text{Sequencing Depth}\right) + \text{Study ID} + \text{CRC}
\end{align}`
and fit associated ZINB models using all data from studies `\(X^{(1)}, \dots, X^{(K)}\)`.

This lets us track `\(P\left(H_{j1} \vert X^{(1)}, \dots, X^{(K)}\right)\)`.

---

### Remaining Studies

This provides a simple summary of how evidence has evolved as more studies are
completed.

.center[
&lt;img src="figures/hypothesis_evolution.png" width = 900/&gt;
]

---

### Remaining Studies

We can focus on some of the more interesting curves, and the low-level data seem
consistent with the high-level summary.

.center[
&lt;img src="figures/hypothesis_evolution_example.png" width = 900/&gt;
]

---

### Bonus: &lt;span style="color:#D94E4E"&gt;Interactive Plot&lt;/span&gt;

We have an experimental interactive visualization that can be used for model criticism. You can check out the code: https://go.wisc.edu/q43ol3

.center[
&lt;a href="http://localhost:5000"&gt;
&lt;img src="figures/scdesigner-interactive.png" width=700/&gt;
&lt;/a&gt;
]

---


### Thank you!

* Lab Members: Margaret Thairu, Hanying Jiang, Shuchen Yan, Mason Garza, Yuliang Peng, and Kaiyan Ma
* Discussions with Jingyi Jessica Li (UCLA)
* Funding: NIGMS R01GM152744.

---

### References

[1] D. L&lt;U+00E4&gt;hnemann, J. K&lt;U+00F6&gt;ster, E. Szczurek, et al. "Eleven
grand challenges in single-cell data science". In: _Genome Biology_ 21
(2020). &lt;https://api.semanticscholar.org/CorpusID:211054289&gt;.

[2] L. Zappia, B. Phipson, and A. Oshlack. "Splatter: simulation of
single-cell RNA sequencing data". In: _Genome Biology_ 18 (2017).
&lt;https://api.semanticscholar.org/CorpusID:9332625&gt;.

[3] D. Song, Q. Wang, G. Yan, et al. "scDesign3 generates realistic in
silico data for multimodal single-cell and spatial omics". In: _Nature
Biotechnology_ (2023), pp. 1-6.
&lt;https://api.semanticscholar.org/CorpusID:258638565&gt;.

---

### References

[4] S. N. Goodman. "Of P-values and Bayes: a modest
proposal." In: _Epidemiology_ 12 3 (2001), pp.  295-7
.
&lt;https://api.semanticscholar.org/CorpusID:41986475&gt;.

[5] E. Pasolli, L. Schiffer, P. Manghi, et al.
"Accessible, curated metagenomic data through
ExperimentHub". In: _Nature Methods_ 14 (2017), pp.
1023-1024.
&lt;https://api.semanticscholar.org/CorpusID:3403081&gt;.

[6] B. J. Callahan, K. Sankaran, J. Fukuyama, et al.
"Bioconductor Workflow for Microbiome Data Analysis:
from raw reads to community analyses". In:
_F1000Research_ 5 (2016).
&lt;https://api.semanticscholar.org/CorpusID:6468345&gt;.

---

### References

[7] G. H. Golub and H. Zha. "Perturbation analysis of
the canonical correlations of matrix pairs". In:
_Linear Algebra and its Applications_ 210 (1994), pp.
3-28.
&lt;https://api.semanticscholar.org/CorpusID:123160725&gt;.

[8] K. Arikawa. "Theoretical framework for analyzing
structural compliance properties of proteins". In:
_Biophysics and Physicobiology_ 15 (2018), pp. 58 -
74.
&lt;https://api.semanticscholar.org/CorpusID:4472297&gt;.

[9] M. A. Hernan and J. M. Robins. _Causal
inference_. En. Boca Raton, FL: CRC Press, Jan. 2024.

[10] J. J. Li. "Using Synthetic Null Data to Enhance
Statistical Rigor in Genomics".
&lt;http://jsb.ucla.edu/sites/default/files/072523_Overton.pdf&gt;.

---

### References

[11] B. Hofner, A. Mayr, and M. Schmid. "gamboostLSS:
An R Package for Model Building and Variable
Selection in the GAMLSS Framework". In: _arXiv:
Computation_ (2014).
&lt;https://api.semanticscholar.org/CorpusID:62302678&gt;.

[12] R. A. Rigby and D. M. Stasinopoulos.
"Generalized additive models for location, scale and
shape". In: _Journal of the Royal Statistical
Society: Series C (Applied Statistics)_ 54 (2005).
&lt;https://api.semanticscholar.org/CorpusID:56076930&gt;.

[13] K. Cao, Y. Hong, and L. Wan. "Manifold alignment
for heterogeneous single-cell multi-omics data
integration using Pamona". In: _Bioinformatics_ 38
(2020), pp. 211 - 219.
&lt;https://api.semanticscholar.org/CorpusID:226959190&gt;.

---

### Verbs: Plot


```r
sample(simulator) |&gt;
  plot_correlations(points = point_opts)
```

&lt;img src="20231206_files/figure-html/correlation_plot-1.png" width="600" style="display: block; margin: auto;" /&gt;

---

### Verbs: &lt;span style="color:#D94E4E"&gt;Mutate&lt;/span&gt;

There are many reasons we might want to alter an initial simulator,
* **Synthetic Nulls**: We can define negative controls to safeguard against false discoveries.
* **Iteration**: We may want to improve the simulator’s goodness of fit by modifying the regression.
* **Power Analysis**: We may want to construct data from alternative experimental designs or biological signal strengths.

---

### &lt;span style="color:#025E73"&gt;Modularity&lt;/span&gt; - Augment

Another way to alter a simulator is to make implicit data explicitly available,
in the spirit of the broom package's `augment`.

For example, these data appear to have mouse-level effects.

.center[
&lt;img src="figures/subject_effect_real.png" width=750/&gt;
]

---

### &lt;span style="color:#025E73"&gt;Modularity&lt;/span&gt; - Augment

1. Unfortunately, none of the packages we use at the marginal estimation step support random effects.

1. We defined an `augment` function that defines a simple proxy,

    * Compute sample-wise low-dimensional multidimensional scaling coordinates `\(z_{i}\)`.
    * For mouse `\(s\)`, condition on its average coordinates: `\(F_{g}\left(y_{ig} \vert x_{i}, \bar{z}_{s\left(i\right)}\right)\)`
    * Use with caution: The `\(z_{i}\)` will be correlated with the diet effect.

.large-code[

```r
exper &lt;- augment_mds(exper, K = 4) |&gt;
  groupwise_average(starts_with("MDS"), MouseID)
```
]

---

### &lt;span style="color:#025E73"&gt;Modularity&lt;/span&gt; - Augment

* Compute sample-wise low-dimensional multidimensional scaling coordinates `\(z_{i}\)`.
* For mouse `\(s\)`, condition on its average coordinates: `\(F_{g}\left(y_{ig} \vert x_{i}, \bar{z}_{s\left(i\right)}\right)\)`
* Use with caution: The `\(z_{i}\)` will be correlated with the diet effect.

.center[
&lt;img src="figures/mouse_effect_comparison.png" width=1000/&gt;
]

---

### Example 1: Multi-omics Power Analysis

1. In practice, most power analyses are univariate, e.g., to guide differential expression.
2. How should we do multivariate power analysis, especially with several tables?

---

### Example 1

Let’s consider the microbiome + metabolome mouse study from [9]. This had 12 samples.
1. How does estimation quality improve with sample size?
2. Is it worth gathering unpaired samples? (mosaic designs)

.center[
&lt;img src="figures/cca_f1000.gif" width=650/&gt;
]

---

### Example 1: Estimate Simulators

1. Fit ZINB models to the microbiome.
2. Fit Gaussian models to the metabolome.
3. Join them using highly regularized covariance estimates.

.center[
  &lt;img src="figures/cca_microbiome_hist.png"/&gt;
]

---

### Part 1: Estimate Simulators

1. Fit ZINB models to the microbiome.
2. Fit Gaussian models to the metabolome.
3. Join them using highly regularized covariance estimates.

.center[
  &lt;img src="figures/cca_metab_hist.png"/&gt;
]

---

### Part 2: Hypothetical Experiments
.pull-left[
1. Across sample sizes `\(n\)`, we simulate new data, fit sparse CCA, and identify a
basis `\(\hat{V}_{n}\)` spanning the `\(K = 5\)` left and right sparse CCA factors
1. We use `\(\hat{V}_{5000}\)` as a reference and compute canonical angles [10] with all smaller sample sizes.
]

.pull-right[
&lt;img src="figures/canonical_angles_paper.png"/&gt;
Figure from [11].
]

---

### Part 2: Hypothetical Experiments

With more samples, the canonical angles shrink predictably.

.center[
&lt;img src="figures/canonical_angles.png" width=750/&gt;
]

---

### Mosaic Case

Here, we have allowed the sample sizes to differ and use KNN imputation to run
CCA on an imputed table.

.center[
&lt;img src="figures/mosaic_example.png" width=750/&gt;
]

---

### Statistics as the Science of Thought Experiments

.pull-left[
1. Precise technology `\(\neq\)` precise thought. The opacity of translating scientific hypotheses into quantitative models arguably contributes to the reproducibility crisis [12; 13]. 

1. Simulation-based methods train researchers in this translation, and
effective interactive computing interfaces can accelerate their adoption.
]

.pull-right[
&lt;img src="figures/somnium.jpeg" width=350/&gt;
]

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
