<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Intervention Analysis with Mirrors</title>
    <meta charset="utf-8" />
    <meta name="author" content="Kris Sankaran" />
    <script src="libs/header-attrs-2.27/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="css/xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


class: title

`\(\def\Dir{\text{Dir}}\)`
`\(\renewcommand{\exp}[1]{\operatorname{exp}\left(#1\right)}\)`
`\(\def\Mult{\text{Mult}}\)`
`\(\def\*#1{\mathbf{#1}}\)`
`\(\def\m#1{\boldsymbol{#1}}\)`
`\(\def\Unif{\text{Unif}}\)`
`\(\def\win{\tilde{w}_{\text{in}}}\)`
`\(\def\reals{\mathbb{R}}\)`
`\(\def\simplex{\Delta}\)`
`\(\def\*#1{\mathbf{#1}}\)`
`\(\def\m#1{\boldsymbol{#1}}\)`
`\(\def\PD{\mathrm{PD}}\)`
`\(\def\FDP{\mathrm{FDR}}\)`
`\(\newcommand\mbb[1]{\mathbb{#1}}\)`
`\(\newcommand\mbf[1]{\mathbf{#1}}\)`
`\(\def\mc#1{\mathcal{#1}}\)`
`\(\def\mrm#1{\mathrm{#1}}\)`
`\(\def\absarg#1{\left|#1\right|}\)`
&lt;div style = "position:fixed; visibility: hidden"&gt;
`$$\require{color}\definecolor{myred}{rgb}{0.705882352941177, 0.341176470588235, 0.36078431372549}$$`
`$$\require{color}\definecolor{mygreen}{rgb}{0.352941176470588, 0.541176470588235, 0.501960784313725}$$`
&lt;/div&gt;

&lt;script type="text/x-mathjax-config"&gt;
MathJax.Hub.Config({
  TeX: {
    Macros: {
      myred: ["{\\color{myred}{#1}}", 1],
      mygreen: ["{\\color{mygreen}{#1}}", 1]
    },
    loader: {load: ['[tex]/color']},
    tex: {packages: {'[+]': ['color']}}
  }
});
&lt;/script&gt;

&lt;style&gt;
.myred {color: #B4575C;}
.mygreen {color: #5A8A80;}
&lt;/style&gt;






## Microbiome Intervention Analysis with `mbtransfer`

&lt;div id="subtitle"&gt;
Kris Sankaran &lt;br/&gt;
Joint work with Pratheepa Jeganathan&lt;br/&gt;
06 | August | 2024 &lt;br/&gt;
Lab: &lt;a href="https://go.wisc.edu/pgb8nl"&gt;go.wisc.edu/pgb8nl&lt;/a&gt; &lt;br/&gt;
&lt;/div&gt;

&lt;div id="subtitle_right"&gt;
Methods for Microbiome Data Analysis&lt;br/&gt;
Joint Statistical Meetings&lt;/br&gt;
Slides: &lt;a href="https://go.wisc.edu/h5ow0o"&gt;go.wisc.edu/h5ow0o&lt;/a&gt;&lt;br/&gt;
Paper: &lt;a href=" https://go.wisc.edu/5x8k89"&gt;go.wisc.edu/5x8k89&lt;/a&gt;
&lt;/div&gt;

&lt;!-- 20 minute talk --&gt;

---


### Motivation: Interaction in Dynamic Communities

We are only beginning to understand microbial community dynamics. This knowledge
is critical for moving from description to design [1; 2].

&lt;img src="figure/curtis-opening.png"/&gt;
&lt;span style="font-size: 18px;"&gt;
John Curtis was a UW - Madison Botany
professor who studied how forest ecosystems change in response to environmental
pressures (and created the Bray-Curtis distance). Figure from [3].
&lt;/span&gt;

---

### Interventions

To disentangle these dynamics, longitudinal experimental designs with natural or
induced interventions have gained popularity. The questions are:

1. **Who**: Which taxa are affected by an intervention?
1. **When**: Are the effects immediate? Lagged? Do they persist?
1. **How**: Are there host factors that mediate the effect?

.center[
  &lt;img src="figure/motivation.png"/&gt;
]

---

class: middle

.center[
## Approach
]

---

### Overview

1. **Transfer Functions**: Model intervention and interaction effects across
time horizons using a high-dimensional linear model.

1. **Mirror Statistics**: Use agreement across data splits to allow false
discovery rate-controlled inference of perturbation effects.

.center[
&lt;img src="figure/approach.png" width=800/&gt;
]

---

### Transfer Functions

.pull-left[
Classical transfer function models are autoregressive models:

`$$y_{t} = \sum_{p = 1}^{P} A_{p} y_{t - p} + \sum_{q = 0}^{Q - 1} B_{q}w_{t - q} + \epsilon_{t}$$`

where `\(y_{t}\)` is the series of interest and `\(w_{t}\)` encodes the intervention.
]


.pull-right[
&lt;img src="figure/intervention-analysis.png" width="500" style="display: block; margin: auto;" /&gt;
&lt;span style="font-size: 18px;"&gt;
In Box and Tiao [4], `\(y_{t}\)` are ozone concentrations and `\(w_{t}\)` are new regulations.
 &lt;/span&gt;

]

---

### Transfer Functions

.pull-left[
1. The pattern of autoregressive coefficients determines `\(y_{t}\)`'s response to
 "pulse" and "step" interventions.
1. Note that, unlike standard generalized Lotka-Volterra models [5; 6], they can model lagged intervention effects.  ]
.pull-right[
&lt;img src="figure/pulse.png" width="400" style="display: block; margin: auto;" /&gt;
]


---

### Transfer Functions for Microbial Communities

We estimate a separate transfer function model `\(f_{j}\)` for each taxon:

`$$\mathbf{y}_{t}^{(i)} = \mathbf{f}\left(\mathbf{Y}^{(i)}_{t - 1}, \mathbf{W}^{(i)}_{t}, \mathbf{z}^{(i)} \right) + \mathbf{\epsilon}_{t}^{(i)}$$`

For each coordinate, we use a regularized linear boosting model [7] with candidate interactions formed from an initial prescreen [8].

|Variable| Interpretation |
|--|---|
| `\(\mathbf{y}_{t}^{(i)}\)` | The microbiome community profiles in subject `\(i\)` at time `\(t\)` |
| `\(\mathbf{Y}^{(i)}_{t - 1} \in \reals\)` | The previous `\(L\)` lags of the `\(i^{th}\)` series. |
| `\(\mathbf{W}^{(i)}_{t - 1} \in \reals\)` | An external series whose influence on the response on the microbiome community we would like to identify. |
| `\(\mathbf{z}^{(i)} \in \reals\)` | Subject-level covariates that can interact with `\(\mathbf{Y}_{t - 1}^{(i)}\)` and `\(\mathbf{W}_{t}^{(i)}\)`|

---

### Mirror Statistics

1. Which of the taxa `\(j\)` are affected by the external perturbation, and how long
do the perturbations last?

2. To gauge statistical significance of effects that are not available in model
coefficients, we can use the approach of Dai et al. [9].
.center[
&lt;img src="figure/trajectories_significance.png" width=700/&gt;
]

---

### General Approach: Mirror Statistics

Dai et al. [9] splits the data into `\(\mathcal{D}^{(1)} = \left(\mathbf{X}^{(1)}, \mathbf{y}^{(1)}\right)\)` and `\(\mathcal{D}^{(2)} = \left(\mathbf{X}^{(2)}, \mathbf{y}^{(2)}\right)\)` and check for agreement in the estimates `\(\hat{\mathbf{\beta}}^{(1)}\)` and `\(\hat{\mathbf{\beta}}^{(2)}\)` across splits.

&lt;img src="figure/mirror-explanation1.png" width=800/&gt;

---

### Test Statistic for Linear Model

To detect whether `\(\beta_{j} \neq 0\)`, we can use the statistic:

.center[
`\(M_{j} = \text{sign}\left(\hat{\beta}^{(1)}_{j}\hat{\beta}^{(2)}_{j}\right)\left[\left|\hat{\beta}_{j}^{(1)}\right| + \left|\hat{\beta}_{j}^{(2)}\right|\right]\)`
]

&lt;img src="figure/mirror-explanation1.png" width=800/&gt;

---

### FDR Estimation

To choose a threshold `\(\tau\)`, estimate the false discovery rate:

.center[
`$$\widehat{\FDP}\left(\tau\right) = \frac{\left|\{j : M_{j} &gt; \tau\}\right|}{\left|\left\{j : M_{j} &lt; -\tau\right\}\right|}$$`
]

&lt;img src="figure/mirror-explanation2.png" width=800/&gt;

---

### FDR Estimation

To choose a threshold `\(\tau\)`, estimate the false discovery rate:

.center[
`$$\widehat{\FDP}\left(\tau\right) = \frac{\left|\{j : M_{j} &gt; \tau\}\right|}{\left|\left\{j : M_{j} &lt; -\tau\right\}\right|}$$`
]

.pull-left[
&lt;img src="figure/mirror-explanation2.png" width=500/&gt;
]
.pull-right[
For FDR control, we require,
* Symmetry of `\(M_{j}\)` under the null
* Independence of `\(M_{j}\)`
]

---

### Partial Dependence

For transfer function models, we need an alternative to `\(\beta_{j}\)`. We estimate the partial dependence [7; 10] of the intervention `\(\mathbf{W}_{t}\)` across time horizons.

.center[
`$$\PD_{j}^{(s)} = \frac{1}{\absarg{\mathcal{D}^{(s)}}}\sum_{d_{t}^{(i)} \in \mathcal{D}^{(s)}}\left[\myred{\hat{f}_{j}^{(s)}\left(\*Y^{(i)}_{t}, \*1_{Q}, \*z^{(i)}\right)} - \mygreen{\hat{f}_{j}^{(s)}\left(\*Y^{(i)}_{t}, \*0_{Q}, \*z^{(i)}\right)}\right].$$`
&lt;img src="figure/pd-1.png" width=650/&gt;
]

---

### Partial Dependence

For transfer function models, we need an alternative to `\(\beta_{j}\)`. We estimate the partial dependence [7; 10] of the intervention `\(\mathbf{W}_{t}\)` across time horizons.

.center[
`$$\PD_{j}^{(s)} = \frac{1}{\absarg{\mathcal{D}^{(s)}}}\sum_{d_{t}^{(i)} \in \mathcal{D}^{(s)}}\left[\myred{\hat{f}_{j}^{(s)}\left(\*Y^{(i)}_{t}, \*1_{Q}, \*z^{(i)}\right)} - \mygreen{\hat{f}_{j}^{(s)}\left(\*Y^{(i)}_{t}, \*0_{Q}, \*z^{(i)}\right)}\right].$$`
&lt;img src="figure/pd-2.png" width=650/&gt;
]

---

### Partial Dependence

For transfer function models, we need an alternative to `\(\beta_{j}\)`. We estimate the partial dependence [7; 10] of the intervention `\(\mathbf{W}_{t}\)` across time horizons.

.center[
`$$\PD_{j}^{(s)} = \frac{1}{\absarg{\mathcal{D}^{(s)}}}\sum_{d_{t}^{(i)} \in \mathcal{D}^{(s)}}\left[\myred{\hat{f}_{j}^{(s)}\left(\*Y^{(i)}_{t}, \*1_{Q}, \*z^{(i)}\right)} - \mygreen{\hat{f}_{j}^{(s)}\left(\*Y^{(i)}_{t}, \*0_{Q}, \*z^{(i)}\right)}\right].$$`
&lt;img src="figure/pd-3.png" width=650/&gt;
]

---

### Partial Dependence Mirrors

Our definition uses `\(PD_{j}^{(s)}\)` in place of `\(\hat{\beta}_{j}^{(s)}\)` from the
usual definition.

`\begin{align*}
	M_{j} = \text{sign}\left(\PD_{j}^{(1)}\PD_{j}^{(2)}\right)\left[\absarg{\PD_{j}^{(1)}} + \absarg{\PD_{j}^{(2)}}\right]
\end{align*}`

---

### Summary

.center[
&lt;img src="figure/Fig1.png"/&gt;
]

---

class: middle

.center[
## Experiments
]

---

### Questions

1. **Forecasting**: How do methods compare in their ability to predict microbial
community response to interventions?

1. **Selection**: How do methods compare in their FDR control and power in
detecting affected taxa?

.center[
&lt;img src="figure/inference-pred.png" width=850/&gt;
]

---

### Simulation Parameters

&lt;img src="figure/phylo.png" width=50/&gt; Phylogenetic correlation: We include correlated taxa that reflect shared evolutionary ancestry.&lt;br/&gt;

&lt;img src="figure/reads.png" width=90/&gt; Sequencing depth: In a typical experiment, not all samples are sequenced to the same read depth.

&lt;img src="figure/dimensionality.png" width=50/&gt; Number of (nonnull) taxa: Dimensionality and the signal enrichment varies across applications.

&lt;img src="figure/normal_curve.png" width=80/&gt; Signal strength: Both inference and forecasting are affected by the strength of the intervention.

&lt;img src="figure/transform.png" width=40/&gt; Normalization strategy: We can apply methods to either raw or transformed data.

---

### Generative Mechanism

Negative binomial models are often good fits to microbiome data [11; 12; 13]. We use an
autoregressive version:

`$$\mathbf{y}_{t}^{(i)}\vert\theta_{t}^{(i)},\mathbf{\varphi},b^{(i)}\sim\text{NB}\left(b^{(i)}\exp{\theta_{t}^{(i)}},\mathbf{\varphi}\right)\\\theta_{t}^{(i)}=\sum_{p=1}^{P}A_{p}\theta_{t-p}^{(i)}+\sum_{q=1}^{Q}\left(B_{q}+C_{q}\odot z^{(i)}\right)\mathbf{w}_{t-q}^{i}+\mathbf{\epsilon}_{t}^{(i)}\\b^{(i)}\sim\Gamma\left(10,\lambda\right)\\\epsilon_{t}^{(i)}\sim\mathcal{N}\left(0,\Sigma\right)$$`

---

### Generative Mechanism

.center[
&lt;img src="figure/simulated_data.png" width=800/&gt;&lt;br/&gt;
]
&lt;span style="font-size: 18px"&gt;
Taxonomic abundances in the simulated negative binomial data, transformed to
taxon-specific quantiles. The intervention windows are surrounded in red. Only
the top three rows have true intervention effects.
&lt;/span&gt;

---

### Baselines

Inference

1. `DESeq2` [14] with main and interaction effects for `\(\mathbf{w}_{t}\)` and `\(\mathbf{z}\)`.
1. Two-Sample `\(t\)`-test comparing the four `\(\mathbf{y}_{t}\)` for the first four `\(t\)` after `\(\mathbf{w}_{t} = 1\)` and with all other `\(\mathbf{y}_{t}\)`

Forecasting

1. `MDSINE2` [15; 16; 17]: A Bayesian nonparametric negative binomial autoregressive model for discovering taxonomic interactions.
1. `fido` [18; 19]: A multinomial regression model for count data that we adpted for forecasting.

---


### Prediction Performance: `\(\alpha = 0.1, \lambda = 0.1\)`

&lt;img src="figure/simulation_results_config1.png" width=900/&gt;

---

### Prediction Performance: `\(\alpha = 0.1, \lambda = 10\)`

&lt;img src="figure/simulation_results_config2.png" width=900/&gt;

---

### Prediction Performance: `\(\alpha = 10, \lambda = 0.1\)`

&lt;img src="figure/simulation_results_config3.png" width=900/&gt;

---

### All Settings

&lt;img src="figure/Fig4.png" width=1100/&gt;

---

### Power and FDR

&lt;img src="figure/Fig6.png" width=900/&gt;

---

class: middle

.center[
## Data Analysis
]

---

### Data Description

.pull-left[
David et al. [20] investigated short-term change in
microbiome composition and function in response to diet perturbations.
]

.pull-right[
&lt;img src="figure/diet-study.png" width=300/&gt;
]

---

### In- and Out-of-Sample Prediction

&lt;img src="figure/Fig7.png"/&gt;
&lt;span style="font-size: 18px;"&gt;
Out-of-sample performance is strongest among highly abundant taxa (top row) and
short-term time lags (left column).
&lt;/span&gt;

---

### Mirror Statistics


&lt;img src="figure/S1_Fig12.png"/&gt;

&lt;span style="font-size: 18px;"&gt;
Boxplots of `\(M_{j}\)` that are larger than zero give evidence for significant intervention effects.
&lt;/span&gt;

---

### Estimates and Data

For each taxon, we can compare the partial dependence profiles when the diet
intervention is or is not applied.

&lt;img src="figure/Fig8.png" width=900/&gt;
&lt;span style="font-size: 18px;"&gt;&lt;br/&gt;
Estimates `\(\mathrm{PD}_{j}\)` (top panel) and underlying data (bottom panel) for taxa significant effects.
&lt;/span&gt;

---

### Takeaways

`mbtransfer` supports FDR-controlled inference in forecasting models used for dynamic microbiome data.

.pull-three-quarters-left[
&lt;img src="figure/Fig1.png" width=700/&gt;
]

.pull-three-quarters-right[
[Paper](https://go.wisc.edu/5x8k89)&lt;br/&gt;
&lt;img src="figure/qr-paper.png" width=80/&gt;
&lt;br/&gt;
&lt;br/&gt;
&lt;br/&gt;
[Package](https://krisrs1128.github.io/mbtransfer)&lt;br/&gt;
&lt;img src="figure/qr-package.png" width=80/&gt;
]

---

class: background-rivers

.center[
### Thank you!
]

* Collaborators: Pratheepa Jeganathan (McMaster University)
* Lab Members: Margaret Thairu, Hanying Jiang, Shuchen Yan, Yuliang Peng, Kai Cui, and Kobe Uko
* Funding: NIGMS R01GM152744

---

class: reference

### References

[1] In: _Nature Microbiology_ 9.7 (Jul. 2024), p. 1633–1633. ISSN:
2058-5276. DOI:
[10.1038/s41564-024-01763-1](https://doi.org/10.1038%2Fs41564-024-01763-1).
URL:
[http://dx.doi.org/10.1038/s41564-024-01763-1](http://dx.doi.org/10.1038/s41564-024-01763-1).

[2] S. Tian et al. "Making gut microbiomes from scratch". In: _Cell
Host &amp;amp; Microbe_ 30.11 (Nov. 2022), p. 1508–1509. ISSN: 1931-3128.
DOI:
[10.1016/j.chom.2022.10.005](https://doi.org/10.1016%2Fj.chom.2022.10.005).
URL:
[http://dx.doi.org/10.1016/j.chom.2022.10.005](http://dx.doi.org/10.1016/j.chom.2022.10.005).

[3] J. T. Curtis. _The vegetation of Wisconsin-ordination of plant
communities_. Chapter 7. Madison, WI: University of Wisconsin Press,
Dec. 1959.

[4] G. E. Box et al. "Intervention analysis with applications to
economic and environmental problems". In: _Journal of the American
Statistical association_ 70.349 (1975), pp. 70-79.

[5] D. Gonze et al. "Microbial communities as dynamical systems". In:
_Current Opinion in Microbiology_ 44 (Aug. 2018), p. 41–49. ISSN:
1369-5274. DOI:
[10.1016/j.mib.2018.07.004](https://doi.org/10.1016%2Fj.mib.2018.07.004).
URL:
[http://dx.doi.org/10.1016/j.mib.2018.07.004](http://dx.doi.org/10.1016/j.mib.2018.07.004).

[6] T. A. Joseph et al. "Compositional Lotka-Volterra describes
microbial dynamics in the simplex". In: _PLOS Computational Biology_
16.5 (May. 2020). Ed. by V. Dakos, p. e1007917. ISSN: 1553-7358. DOI:
[10.1371/journal.pcbi.1007917](https://doi.org/10.1371%2Fjournal.pcbi.1007917).
URL:
[http://dx.doi.org/10.1371/journal.pcbi.1007917](http://dx.doi.org/10.1371/journal.pcbi.1007917).

[7] J. H. Friedman. "Greedy function approximation: A gradient
boosting machine." In: _The Annals of Statistics_ 29.5 (Oct. 2001).
ISSN: 0090-5364. DOI:
[10.1214/aos/1013203451](https://doi.org/10.1214%2Faos%2F1013203451).
URL:
[http://dx.doi.org/10.1214/aos/1013203451](http://dx.doi.org/10.1214/aos/1013203451).

[8] G. Thanei et al. "The xyz algorithm for fast interaction search
in high-dimensional data". In: _Journal of Machine Learning Research_
19.37 (2018), pp. 1-42. URL:
[http://jmlr.org/papers/v19/16-515.html](http://jmlr.org/papers/v19/16-515.html).

[9] C. Dai et al. "False Discovery Rate Control via Data Splitting".
In: _Journal of the American Statistical Association_ 118.544 (May.
2022), p. 2503–2520. ISSN: 1537-274X. DOI:
[10.1080/01621459.2022.2060113](https://doi.org/10.1080%2F01621459.2022.2060113).
URL:
[http://dx.doi.org/10.1080/01621459.2022.2060113](http://dx.doi.org/10.1080/01621459.2022.2060113).

[10] P. Biecek. "DALEX: Explainers for Complex Predictive Models in
R". In: _Journal of Machine Learning Research_ 19.84 (2018), pp. 1-5.
URL:
[http://jmlr.org/papers/v19/18-416.html](http://jmlr.org/papers/v19/18-416.html).

[11] X. Zhang et al. "Negative binomial mixed models for analyzing
microbiome count data". In: _BMC Bioinformatics_ 18.1 (Jan. 2017).
ISSN: 1471-2105. DOI:
[10.1186/s12859-016-1441-7](https://doi.org/10.1186%2Fs12859-016-1441-7).
URL:
[http://dx.doi.org/10.1186/s12859-016-1441-7](http://dx.doi.org/10.1186/s12859-016-1441-7).

[12] S. Jiang et al. "A Bayesian zero-inflated negative binomial
regression model for the integrative analysis of microbiome data".
In: _Biostatistics_ 22.3 (Dec. 2019), p. 522–540. ISSN: 1468-4357.
DOI:
[10.1093/biostatistics/kxz050](https://doi.org/10.1093%2Fbiostatistics%2Fkxz050).
URL:
[http://dx.doi.org/10.1093/biostatistics/kxz050](http://dx.doi.org/10.1093/biostatistics/kxz050).

[13] A. K. Mishra et al. "Negative binomial factor regression with
application to microbiome data analysis". In: _Statistics in
Medicine_ 41.15 (Apr. 2022), p. 2786–2803. ISSN: 1097-0258. DOI:
[10.1002/sim.9384](https://doi.org/10.1002%2Fsim.9384). URL:
[http://dx.doi.org/10.1002/sim.9384](http://dx.doi.org/10.1002/sim.9384).

---

class: reference

### References

[14] M. I. Love et al. "Moderated estimation of fold change and
dispersion for RNA-seq data with DESeq2". In: _Genome Biology_ 15.12
(Dec. 2014). ISSN: 1474-760X. DOI:
[10.1186/s13059-014-0550-8](https://doi.org/10.1186%2Fs13059-014-0550-8).
URL:
[http://dx.doi.org/10.1186/s13059-014-0550-8](http://dx.doi.org/10.1186/s13059-014-0550-8).

[15] V. Bucci et al. "MDSINE: Microbial Dynamical Systems INference
Engine for microbiome time-series analyses". In: _Genome Biology_
17.1 (Jun. 2016). ISSN: 1474-760X. DOI:
[10.1186/s13059-016-0980-6](https://doi.org/10.1186%2Fs13059-016-0980-6).
URL:
[http://dx.doi.org/10.1186/s13059-016-0980-6](http://dx.doi.org/10.1186/s13059-016-0980-6).

[16] T. Gibson et al. "Robust and Scalable Models of Microbiome
Dynamics". In: _Proceedings of the 35th International Conference on
Machine Learning_. Ed. by J. Dy and A. Krause. Vol. 80. Proceedings
of Machine Learning Research. PMLR, 2018, pp. 1763-1772. URL:
[https://proceedings.mlr.press/v80/gibson18a.html](https://proceedings.mlr.press/v80/gibson18a.html).

[17] T. E. Gibson et al. "Microbial dynamics inference at
ecosystem-scale".  (Dec. 2021). DOI:
[10.1101/2021.12.14.469105](https://doi.org/10.1101%2F2021.12.14.469105).
URL:
[http://dx.doi.org/10.1101/2021.12.14.469105](http://dx.doi.org/10.1101/2021.12.14.469105).

[18] J. D. Silverman et al. "Dynamic linear models guide design and
analysis of microbiota studies within artificial human guts".  (Apr.
2018). DOI: [10.1101/306597](https://doi.org/10.1101%2F306597). URL:
[http://dx.doi.org/10.1101/306597](http://dx.doi.org/10.1101/306597).

[19] J. D. Silverman et al. "Bayesian Multinomial Logistic Normal
Models through Marginally Latent Matrix-T Processes". In: _Journal of
Machine Learning Research_ 23.7 (2022), pp. 1-42. URL:
[http://jmlr.org/papers/v23/19-882.html](http://jmlr.org/papers/v23/19-882.html).

[20] L. A. David et al. "Diet rapidly and reproducibly alters the
human gut microbiome". In: _Nature_ 505.7484 (Dec. 2013), p. 559–563.
ISSN: 1476-4687. DOI:
[10.1038/nature12820](https://doi.org/10.1038%2Fnature12820). URL:
[http://dx.doi.org/10.1038/nature12820](http://dx.doi.org/10.1038/nature12820).

---

### Phylogenetic Covariance

.center[
&lt;img src="figure/phylogenetic_covariance.png" width=700/&gt;
]

---

### Figure Attributions

math function by Ralf Schmitzer from &lt;a href="https://thenounproject.com/browse/icons/term/math-function/" target="_blank" title="math function Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

biological tree by Leslie Coonrod from &lt;a href="https://thenounproject.com/browse/icons/term/biological-tree/" target="_blank" title="biological tree Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

standard normal distribution by Nick Taras from &lt;a href="https://thenounproject.com/browse/icons/term/standard-normal-distribution/" target="_blank" title="standard normal distribution Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

microbiome by Stefanie Peschel from &lt;a href="https://thenounproject.com/browse/icons/term/microbiome/" target="_blank" title="microbiome Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

---
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
