<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Modular software for mediation analysis of microbiome data</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs-2.29/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="css/xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


class: title

#  Modular software for mediation analysis of microbiome data

&lt;style&gt;
.slide-background {
    background: url("figures/cover.png") no-repeat center center;
    background-size: cover;
    opacity: 0.5;
}
&lt;/style&gt;

&lt;div id="subtitle_left"&gt;
Slides: &lt;a href="https://go.wisc.edu/5ms623"&gt;go.wisc.edu/5ms623&lt;/a&gt;&lt;br/&gt;
Package: &lt;a href="https://go.wisc.edu/830110"&gt;go.wisc.edu/830110&lt;/a&gt;&lt;br/&gt;
Lab: &lt;a href="https://measurement-and-microbes.org"&gt;measurement-and-microbes.org&lt;/a&gt; &lt;br/&gt;
&lt;/div&gt;
&lt;div id="subtitle_right"&gt;
Kris Sankaran &lt;br/&gt;
&lt;a href="https://www.birs.ca/events/2025/5-day-workshops/25w5324/"&gt;Novel Approaches for Multi-Omics&lt;/a&gt;&lt;br/&gt;
17 | July | 2025 &lt;br/&gt;
&lt;/div&gt;

&lt;!-- 25 minute talk (including Q&amp;A) --&gt;



`\(\def\Gsn{\mathcal{N}}\)`
`\(\def\Dir{\text{Dir}}\)`
`\(\def\Mult{\text{Mult}}\)`
`\(\def\diag{\text{diag}}\)`
`\(\def\*#1{\mathbf{#1}}\)`
`\(\def\Scal{\mathcal{S}}\)`
`\(\def\exp#1{\text{exp}\left(#1\right)}\)`
`\(\def\logit#1{\text{logit}\left(#1\right)}\)`
`\(\def\absarg#1{\left|#1\right|}\)`
`\(\def\E{\mathbb{E}} % Expectation symbol\)`
`\(\def\Earg#1{\E\left[{#1}\right]}\)`
`\(\def\P{\mathbb{P}} % Expectation symbol\)`
`\(\def\Parg#1{\P\left[{#1}\right]}\)`
`\(\def\m#1{\boldsymbol{#1}}\)`
`\(\def\Unif{\text{Unif}}\)`
`\(\def\win{\tilde{w}_{\text{in}}}\)`
`\(\def\reals{\mathbb{R}}\)`
`\(\newcommand{\wout}{\tilde w_{\text{out}}}\)`


---

### Microbiome as Mediator

We're just beginning to understand how the microbiome mediates the relationship
between environmental exposures and human health.

.center[
&lt;div class="caption"&gt;
&lt;img src="figures/akkermansia_cholesterol.jpg" width=540/&gt;&lt;br/&gt;
Figure adapted from [1].
&lt;/div&gt;
]

For example, _Akkermansia muciniphila_ can help regulate cholesterol levels by producing
proteins that activate important pathways for cholesterol absorption and
metabolism [2].

---

### Microbiome as Mediator

We're just beginning to understand how the microbiome mediates the relationship
between environmental exposures and human health.

.center[
  &lt;div class="caption"&gt;
&lt;img src="figures/chemotherapy.png"/&gt;
Figure adapted from [3].
&lt;/div&gt;
]

Chemotherapy-induced microbiome changes can worsen adverse events, like
diarrhea and mucositis, limiting treatment dosage and duration 
[4; 5].

---


### Microbiome as Mediator

We're just beginning to understand how the microbiome mediates the relationship
between environmental exposures and human health.

&lt;br/&gt;

.center[
  &lt;div class="caption"&gt;
&lt;img src="figures/examples-dag.png" width=1300/&gt;
&lt;/div&gt;
]

---

### Causal Inference Formalism

Causal mediation analysis is often a good match for carrying out these studies.
Traditional methods from social science and epidemiology need to be adapted to
reflect properties of microbiome data.

.center[
&lt;img src="figures/mediation-dag-terms.png" width=600/&gt;
]

---

### Causal Inference Formalism

Causal mediation analysis is often a good match for carrying out these studies.
Traditional methods from social science and epidemiology need to be adapted to
reflect properties of microbiome data.

.center[
&lt;img src="figures/mediation-dag.svg" width=500/&gt;
]

---

### Causal Inference Formalism

Causal mediation analysis is often a good match for carrying out these studies.
Traditional methods from social science and epidemiology need to be adapted to
reflect properties of microbiome data.

.center[
&lt;img src="figures/mediation-dag.svg" width=500/&gt;
]

---

### Counterfactual Notation

Under the generalized causal mediation analysis framework, we imagine
counterfactuals for both mediators and outcomes, depending on treatments that we
may never have seen.

_Mediator under treatment `\(t\)`_: 
`\begin{align*}
M\left(t\right)
\end{align*}`

_Outcome under treatment combination_ `\(\left(t, t'\right)\)`:
`\begin{align*}
Y\left(t, M\left(t'\right)\right)
\end{align*}`

Note that this is not observable whenever `\(t \neq t'\)`, but we can reason about
it abstractly.

---

### Direct vs. Indirect Effects

Suppose that the treatments belong to two groups, `\(T \in \{0, 1\}\)`. Then, direct
and indirect effects are defined as

`\begin{align*}
\bar{\delta} = \frac{1}{2} \sum_{t=0}^{1} \mathbb{E}\{Y(t, M(1)) - Y(t, M(0))\}\\
\bar{\zeta} = \frac{1}{2} \sum_{t'=0}^{1} \mathbb{E}\{Y(1, M(t')) - Y(0, M(t'))\}
\end{align*}`

When many mediators are present, we may also want to know:

`\begin{align*}
\bar{\omega}_k = \frac{1}{2} \sum_{t'=0}^{1} \mathbb{E}\{Y(t', M_k(1), M_{-k}(t')) - Y(t', M_k(0), M_{-k}(t'))\}
\end{align*}`

Under appropriate sequential ignorability assumptions, these effects can be
identified.

---

### Geometric Interpretation

&lt;img src="figures/geometric_1.svg" width=800/&gt;

---

### Geometric Interpretation

&lt;img src="figures/geometric_2.svg" width=800/&gt;

---

### Geometric Interpretation

&lt;img src="figures/geometric_3.svg" width=800/&gt;

---

### Integration Challenge

Our interest in this project came from an attempt to re-analyze a dataset about
the gut brain axis. Participants were randomized into either a mindfulness
intervention or a control group. In this case, the microbiome is an outcome, and
the survey measurements are mediators.

---

### Integration Challenge

A similar issue arises when applying mediation to support metabolomic
integration. This is a quote from a study on the microbiome and metabolome of
inflammatory bowel disease [6]:

&lt;div style="font-size: 20px"&gt;
&gt; The CD-associated compounds eicosatrienoic (ETA) and docosapentaenoic (DPA)
acid were involved in negative associations with control-associated species and
positive associations with IBD-associated species. ETA and DPA are
polyunsaturated long-chain fatty acids (PUFAs) [which] possess bactericidal
activity by virtue of their hydrophobic nature and potential to disrupt
bacterial cell membranes. 
&lt;/div&gt;

As in the previous example, we can imagine a mediation analysis with a
high-dimensional microbiome outcome.

---

.center[
## Package Design
]

---

### Univariate Mediation Analysis

The most straightforward implementation of mediation analysis with univariate
mediators and outcomes is a two-step linear regression:

`\begin{align*}
y_i = \beta_t t_{i} + \beta_{m}m_i + \epsilon_i^y\\
m_{i} = \alpha t_i + \epsilon_{i}^{m_{i}}
\end{align*}`

In this case, the direct and indirect effects are straightforward:

Direct effect: `\(\beta_t\)`
Indirect effect: `\(\alpha \beta_m\)`

---

### Code Interface

We could imagine writing a function to implement this analysis.

.large-code[

``` r
model &lt;- mediation(y, m, t)
effects(model)
```
]

But how can we adopt the more general counterfactual perspective, without
restricting ourselves to linear models?

---

### Code Interface

The `mediation` package [7; 8]
solves this problem by instead creating an interface like:

.large-code[

``` r
f1 &lt;- lm(y ~ t + m)
f2 &lt;- lm(m ~ t)
model &lt;- mediation(f1, f2)
effects(model)
```
]

We can now accommodate any mediation or outcome model, not just linear ones!
Indeed, it's easy to use `glm` in the `mediation` package.

---

### Multimedia Interface

`multimedia` generalizes this interface for multivariate responses and outcomes
appropriate to microbiome data. For example, for the mindfulness study, we use a
logistic-normal multinomial model for the outcomes and lasso regression for the
mediators.

.large-code[

``` r
model &lt;- multimedia(
    exper, # dataset
    lnm_model(), # outcome model
    glmnet_model(lambda = 0.5, alpha = 0) # mediation model
)
```
]

We fit each model separately, but this can still accommodate a few commonly used
approaches.

---

### Data Formats


Besides modeling, `multimedia` has data structures that make it easier to
manipulate real and counterfactual data. For example, we can use `tidy`
selection syntax to categorize variables.


``` r
head(ibd_data, 4)
```

```
## # A tibble: 4 × 341
##   m0031_phenyllactate m0045_azelate m0171_palmitic_acid m0219_2_hydroxyhexadecanoate m0244_alpha_linoleni…¹ m0246_linoleic_acid m0250_oleic_acid m0256_stearic_acid m0354_arachidonic_acid m0393_1213_diHOME
##                 &lt;dbl&gt;         &lt;dbl&gt;               &lt;dbl&gt;                        &lt;dbl&gt;                  &lt;dbl&gt;               &lt;dbl&gt;            &lt;dbl&gt;              &lt;dbl&gt;                  &lt;dbl&gt;             &lt;dbl&gt;
## 1                9.44          7.18                8.26                        10.0                    7.02                8.86             9.28               8.13                   7.92              6.23
## 2                8.92          6.47                8.19                         9.57                   6.29               10.4             11.1                9.10                   5.47              5.10
## 3                6.36          8.16                7.39                        10.4                    7.49                8.88             8.65               7.60                   8.74              4.82
## 4                8.24          5.46                9.44                         9.83                   5.78                8.53             8.68               8.83                   6.54              6.94
## # ℹ abbreviated name: ¹​m0244_alpha_linolenic_acid
## # ℹ 331 more variables: m0472_docosahexaenoic_acid &lt;dbl&gt;, m0731_lithocholic_acid &lt;dbl&gt;, m0830_hyodeoxycholateursodeoxycholate &lt;dbl&gt;, m1129_chenodeoxycholate &lt;dbl&gt;,
## #   m1360_chenodeoxycholatedeoxycholate &lt;dbl&gt;, m0833_deoxycholic_acid &lt;dbl&gt;, m0900_ketodeoxycholate &lt;dbl&gt;, m0930_cholate &lt;dbl&gt;, m1258_cholate &lt;dbl&gt;, m1303_lithocholate &lt;dbl&gt;, m1403_cholate &lt;dbl&gt;,
## #   m0253_sphingosine &lt;dbl&gt;, m0906_threosphingosine &lt;dbl&gt;, m0998_threosphingosine &lt;dbl&gt;, m0927_piperine &lt;dbl&gt;, m0264_linoleoyl_ethanolamide &lt;dbl&gt;, m0339_linoleoyl_ethanolamide &lt;dbl&gt;,
## #   m0410_C180e_MAG &lt;dbl&gt;, m0416_cholesterol &lt;dbl&gt;, m0467_C180_MAG &lt;dbl&gt;, m0478_cholestenone &lt;dbl&gt;, m0554_cholestenone &lt;dbl&gt;, m0878_C160_LPC &lt;dbl&gt;, m0915_C200_LPE &lt;dbl&gt;, m0949_C180_LPC &lt;dbl&gt;,
## #   m1271_C342_DAG &lt;dbl&gt;, m1291_C342_DAG &lt;dbl&gt;, m1277_C341_DAG &lt;dbl&gt;, m1351_C364_DAG &lt;dbl&gt;, m1367_C364_DAG &lt;dbl&gt;, m1358_C363_DAG &lt;dbl&gt;, m1373_C363_DAG &lt;dbl&gt;, m1364_C362_DAG &lt;dbl&gt;, m1377_C362_DAG &lt;dbl&gt;,
## #   m1478_C182_CE &lt;dbl&gt;, m1504_C182_CE &lt;dbl&gt;, m1514_C181_CE &lt;dbl&gt;, m1657_C160_SM &lt;dbl&gt;, m2064_C501_TAG &lt;dbl&gt;, m2081_C501_TAG &lt;dbl&gt;, m2075_C502_TAG &lt;dbl&gt;, m2138_C525_TAG &lt;dbl&gt;, m2132_C524_TAG &lt;dbl&gt;, …
```

These are the data from the motivating IBD study as a raw `data.frame`.

---

### Data Formats

Besides modeling, `multimedia` has data structures that make it easier to
manipulate real and counterfactual data. For example, we can use `tidy`
selection syntax to categorize variables.


``` r
library(multimedia)
mediation_data(
  ibd_data,
  matches("^m[0-9]{4}"), # outcomes
  "Study.Group", # treatments
  starts_with("g") # mediators
)
```

```
## [Mediation Data] 
## 220 samples with measurements for, 
## 1 treatment: Study.Group 
## 173 mediators: gSutterella, gMarvinbryantia, ... 
## 155 outcomes: m0031_phenyllactate, m0045_azelate, ...
```

---

### Effect Estimates

Once we have fitted mediator `\(\hat{M}\)` and outcome `\(\hat{Y}\)` models, we can
estimate the associated direct and indirect effects.

`\begin{gathered}
\widehat{\bar{\delta}}=\frac{1}{2} \sum_{t=0}^1 \sum_{i=1}^n \widehat{Y}_i\left(t, \widehat{M}_i(1)\right)-\widehat{Y}_i\left(t, \widehat{M}_i(0)\right), \\
\widehat{\bar{\zeta}}=\frac{1}{2} \sum_{t^{\prime}=0}^1 \sum_{i=1}^n \widehat{Y}_i\left(1, \widehat{M}_i\left(t^{\prime}\right)\right)-\widehat{Y}_i\left(0, \widehat{M}_i\left(t^{\prime}\right)\right), \\
\widehat{\bar{\omega}}=\frac{1}{2} \sum_{t^{\prime}=0}^1 \sum_{i=1}^n\left\{\widehat{Y}_i\left(t^{\prime}, \widehat{M}_{i k}(1), \widehat{M}_{i,-k}\left(t^{\prime}\right)\right)-\widehat{Y}_i\left(t^{\prime}, \widehat{M}_{i k}(0), \widehat{M}_{i,-k}\left(t^{\prime}\right)\right)\right\}
\end{gathered}`

---

### Bootstrap

We can gauge uncertainty in the estimated effects by re-estimating both mediator
and outcome models on bootstrap resampled versions of the original data.  Here
are the metabolites with the largest indirect and direct effects in the IBD
example. 
.center[
&lt;img src="figures/lasso_bootstrap_vis-1.png" width=800/&gt;
]

---

### Model Alteration

The package provides syntax for altering models after they've been estimated.
We can set specific coefficients to zero or re-estimate with new model
specifications.

.center[
&lt;img src="figures/visualize_long-1.png"/&gt;
]

---


``` r
fit &lt;- multimedia(exper, outcome = rf_model(num.trees = 1e3)) |&gt;
    estimate(exper)

altered_m &lt;- nullify(fit, "T-&gt;M") |&gt;
    estimate(exper)
altered_ty &lt;- nullify(fit, "T-&gt;Y") |&gt;
    estimate(exper)
```
.center[
&lt;img src="figures/visualize_long-1.png"/&gt;
]

---

### Synthetic Null Mediators

Here is the same principle applied to the mindfulness study. The synthetic null
survey responses have been generated from mediation models with `\(\alpha = 0\)`.

.center[
&lt;img src="figures/mindfulness-altered.png" width=780/&gt;&lt;br/&gt;
&lt;span style="font-size: 24px;"&gt;
The middle panel comes from a synthetic null: `\(T \nrightarrow M \to Y\)`.
&lt;/span&gt;
]

---

These are analogous comparisons for the simulated microbiomes. Sampling from the
fitted mediation models can help with model checking.

.center[
&lt;img src="figures/alteration_plot-1.png" width=650/&gt;&lt;br/&gt;
]

---

### Sensitivity Analysis

1. Mediation analysis relies on untestable assumptions. Nonetheless, the
consequences of their violation can be explored using sensitivity analysis.

1. We provide methods for simulation counterfactual mediator and outcome
variables with correlated noise terms, capturing the situation where the
assumption of no pretreatment confounding is violated.

---

### Sensitivity Analysis

For example, once we have fit mediation and outcome models, we can simulate
according to:

`\begin{align*}
Y^*(t, m)=\hat{Y}(t, m)+\epsilon^y \\
M^*(t)=\hat{M}(t)+\epsilon^m .
\end{align*}`

A caveat is that this only makes sense for continuous outcomes and mediators.

---

### Sensitivity Analysis

We draw `\(\left(\epsilon^y, \epsilon^m\right)\)` from a gaussian with mean zero and covariance,
`\begin{align*}
\Sigma(\rho, G):=\left(\begin{array}{cc}
\operatorname{diag}\left(\hat{\sigma}_M^2\right) &amp; \rho \hat{\sigma}_M \hat{\sigma}_Y^{\top} \odot \mathbf{1}_G \\
\rho \hat{\sigma}_Y \hat{\sigma}_M^{\top} \odot \mathbf{1}_G^{\top} &amp; \operatorname{diag}\left(\hat{\sigma}_Y^2\right)
\end{array}\right)
\end{align*}`
where `\(\mathbf{1}_{G} \in \{0, 1\}^{K \times J}\)` is an indicator over
mediator-outcome pairs `\(G\)` over which to test sensitivity.

---

.center[
## Example
]

---

### Study Background

1. The study  attempted an integrative analysis to discover
metabolite and taxonomic markers that might have diagnostic or therapeutic
potential.

1. They gathered untargeted metabolomics + whole genome sequencing microbiome
data on a cohort of 220 patients with inflammatory bowel disease (Ulcerative
Colitis and Crohn's Disease). 

1. This resulted in 8.8K and 11.7K metabolite and genus features, respectively,
all available on the [Curated Metabolome-Microbiome repository](https://github.com/borenstein-lab/microbiome-metabolome-curated-data/tree/main).

---

### Model Setup

1. We applied centered log-ratio and `\(\log\left(1 + x\right)\)` transformations to
the microbiome and metabolome data, respectively. Both were then filtered to
between 150 - 200 of the most abundant features.

1. We treat the microbiome as the mediators and the metabolome as the outcome.
This direction could be debated, but follows the discussion in [6]. 

1. Microbe abundance depends only on treatment status. Each metabolite's
abundance is a sparse linear function of a few microbes.

.large-code[

``` r
model &lt;- multimedia(exper, glmnet_model(lambda = 0.1))
```
]

---

### Indirect Effects

This multidimensional scaling (MDS) is based on only microbiome data. Point size
reflects metabolite abundances. Associations between metabolites and the MDS
appear only for outcomes with large indirect effects.

.center[
&lt;img src="figures/indirect_effects_metabolites.png" width=900/&gt;
]

---

### Pathwise Indirect Effects

The top pathwise indirect effects are similar to the geometric interpretation we
saw in the introduction!

.center[
&lt;img src="figures/indirect_effects_pathwise-metabolites.png"/&gt;
]

---

### Sensitivity Analysis

In this sensitivity analysis, we simulate unmeasured confounding between the
abundances of the Enterocloster genus (mediator) and the metabolites (outcomes)
`hydrocinnamic_acid`, `lithocholate`, and `arginine`.

.center[
&lt;img src="figures/indirect_effects_sensitivity.png"/&gt;
]

---

### Takeaways

1. Mediation analysis is a useful framework for relating environment, host, and
microbiome features. More generally, counterfactual language can guide
multi-omics data integration.

1. Modular software design can ensure that a few core statistical components can
be applied to a wide range of study applications.


---

### Thank you!

Paper: [go.wisc.edu/ebm917](https://go.wisc.edu/ebm917)
Package: [go.wisc.edu/830110](https://go.wisc.edu/830110)

* Contact: ksankaran@wisc.edu
* Lab Members: Margaret Thairu, Shuchen Yan, Yuliang Peng, Helena Huang
* Funding: NIGMS R01GM152744, NIAID R01AI184095

---

class: reference

### References

[1] E. A. Franzosa et al. "Gut microbiome structure and metabolic activity in inflammatory bowel disease". In: _Nature Microbiology_ 4.2 (Dec. 2018), p. 293–305. ISSN: 2058-5276.
DOI: [10.1038/s41564-018-0306-4](https://doi.org/10.1038%2Fs41564-018-0306-4). URL: [http://dx.doi.org/10.1038/s41564-018-0306-4](http://dx.doi.org/10.1038/s41564-018-0306-4).

---

### Example Model: Logistic-Normal Multinomial

The logistic-normal multinomial (LNM) model has the form:

.pull-left[
`\begin{align*}
y_{i} \sim \Mult\left(N_{i}, \varphi^{-1}\left(z_{i}^{T}\beta\right)\right) \\
\beta \sim \Gsn\left(0, \diag\left(\sigma_{k}^{2}\right)\right)
\end{align*}`
where `\(\varphi^{-1}\left(z\right) \propto\left(\exp{z_{1}}, \dots, \exp{z_{K-1}}, 1\right)\)`
]

.pull-right[
&lt;img src="figures/lnm.svg" style="display: block; margin: auto;" /&gt;
]

---

### Alternative: Hurdle Model

The interface makes it easy to swap in new models. For example, here we replaced
the lasso regression on log-normalized metabolite abundances with a hurdle model
that directly models zero-inflated nonnegative data.

.large-code[

``` r
model &lt;- multimedia(
    exper2, # version without log transformation
    brms_model(family = hurdle_lognormal()) # new outcome model
) |&gt;
    estimate(exper2)
```
]

---

### Alternative: Hurdle Model

.center[
&lt;img src="figures/hurdle_model_indirect.png" width=700/&gt;
]
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
  "highlightStyle": "github",
  "highlightLines": true,
  "ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
