<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Semisynthetic Controls for Cancer Microbiome Analysis</title>
    <meta charset="utf-8" />
    <script src="libs/header-attrs-2.29/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link rel="stylesheet" href="css/xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


class: title

# Semisynthetic Controls for Cancer Microbiome Analysis

&lt;style&gt;
.slide-background {
    background: url("figures/cover.png") no-repeat center center;
    background-size: cover;
    opacity: 0.5;
}
&lt;/style&gt;

&lt;div id="subtitle_left"&gt;
Slides: &lt;a href="https://go.wisc.edu/02hl09"&gt;go.wisc.edu/02hl09&lt;/a&gt;&lt;br/&gt;
Paper: &lt;a href="https://go.wisc.edu/p12o8w"&gt;go.wisc.edu/p12o8w&lt;/a&gt;&lt;br/&gt;
Lab: &lt;a href="https://measurement-and-microbes.org"&gt;measurement-and-microbes.org&lt;/a&gt; &lt;br/&gt;
&lt;/div&gt;
&lt;div id="subtitle_right"&gt;
Kris Sankaran &lt;br/&gt;
&lt;a href="https://symposium2025.icsa.org/"&gt;ICSA 2025&lt;/a&gt;&lt;br/&gt;
17 | June | 2025 &lt;br/&gt;
&lt;/div&gt;

&lt;!-- 20 minute talk --&gt;



---

exclude: true

### Outline

1. Cancer and the microbiome
  - Recent developments
  - Statistical challenges

1. Semisynthetic Data
  - Conceptual overview
  - Analysis examples

---

.center[
## Cancer and the Microbiome
]

---

### Overview

The relationship between the microbiome and cancer is well-established but
complex [1; 2; 3; 4].

1. Microbes can alter (1) gene pathways, (2) immunological activity, and (3)
metabolic environments in ways that can accelerate or protect against cancer
development or progression.

1. The microbiome, unlike the human genome, is potentially alterable.
Researchers are studying ways it can be optimized to improve treatment.

---

### Carcinogenesis - Direct

.pull-left[
Some species can promote cancer. For example, [5] found enrichment of _F. nucleatum_ in matched colorectal
cancer vs. healthy tissue. These bacteria can bind to cell surfaces and activate
cancer-inducing cytokines.
]

.pull-right[
  &lt;div class="caption"&gt;
  &lt;img src="figures/fnulceatum_cartoon.jpg" width=250/&gt;
  &lt;br/&gt;
  A summary of pathways triggered by an oncogenic microbe from [4].
  &lt;/div&gt;
]

---

### Carcinogenesis - Direct

.pull-left[
Some species can promote cancer. For example, [5] found enrichment of _F. nucleatum_ in matched colorectal
cancer vs. healthy tissue. These bacteria can bind to cell surfaces and activate
cancer-inducing cytokines.
]

.pull-right[
  &lt;div class="caption"&gt;
  &lt;img src="figures/fnucleatum_matched_pairs.png" width=575/&gt;
  &lt;br/&gt;
  The original data presented in [5]. Each bar is
  a patient, and the y-value is the estimated relative change in abundance
  between disease vs. matched healthy tissue.
  &lt;/div&gt;
]

---

### Carcinogenesis - Indirect

.pull-left[
In a mouse model of liver cancer, [6] found that the microbiome
can produce metabolites that convert primary to secondary bile acids. 

The primary bile acids up-regulate CXCL16 production, which protects against
metastasis -- the microbiome weakens this effect.
]

.pull-right[
  &lt;div class="caption"&gt;
  &lt;img src="figures/bile_acids.jpeg" width=320/&gt;&lt;br/&gt;
  Figure from [7].
  &lt;/div&gt;
]

---

### Chemotherapy

Microbiome changes induced by chemotherapy can exacerbate adverse events, like
diarrhea and mucositis, which then limit the dosage and duration of chemotherapy
treatments [8; 9; 10].
&lt;br/&gt;
&lt;br/&gt;

.center[
  &lt;div class="caption"&gt;
&lt;img src="figures/chemotherapy.png"/&gt;
Figure adapted from [11].
&lt;/div&gt;
]

---

### Immune Checkpoint Inhibitors

.pull-left[
Gut microbiome composition has been associated with the effectiveness of immune
checkpoint inhibitors [12; 13]. This fact has
been exploited in clinical trials where participants undergo FMT before
immunotherapy treatment.
]

.pull-right[
  &lt;div class="caption"&gt;
  &lt;img src="figures/immunotherapy.png"/&gt;
  Figure adapted from [12].
  &lt;/div&gt;
]

---

### Drug Delivery

There are bacteria that are known to home in on tumor environments. The study 
[14] engineered a system for delivering them through
alternating periods of periodic growth and lysis.

.center[
  &lt;div class="caption"&gt;
  &lt;img src="figures/tumor_bacteria_delivery.png"/&gt;
  Figure adapted from [14].
  &lt;/div&gt;
]

---

### Statistical Challenges

&gt; Technically, such big data harmonization and integration is far from trivial, while "in-house" meethodologies, even when applied to the same datasets, may lead to nonoverlapping conclusions"

  -- _Microbiome and cancer_ [4]

&gt; Overall, there is not a great deal of overlap between specific bacterial taxa associated with response [to therapy]...differences may be related to several different factors--including differences in techniques used to analyze samples...suggesting the importance of developing standardized approaches for microbiome analysis.

  -- _The influence of the gut microbiome on cancer, immunity, and cancer immunotherapy_ [1]

---

.center[
## Simulation 
]

---

### Data Analysis Controversy 

.center[
    &lt;img src="figures/retraction.png"/&gt;
]

In June 2024, _Nature_ retracted a paper [15] the claimed
identify microbiome signatures of cancer. This came after one year's worth of
debate [16; 17]
about the data analysis.

---

### Data Analysis Controversy 

.center[
    &lt;img src="figures/retraction.png"/&gt;
]

The "disease signature" was an artifact resulting from the use of a batch effect
correction method.  Before we can understand the nuances of the story, we need
to learn about batch effects and correction methods.

---

### Simulation to Resolve Controversy

Gerry Tonkin-Hill has an excellent re-analysis [18] of the data from [15] which sheds light what were likely the source of the phantom
signals. The first part is a simulation.

.pull-left[
&lt;img src="figures/gth_sim1.png"/&gt;
]
.pull-right[
&lt;img src="figures/gth_sim2.png"/&gt;
]

---

### Why Simulate?

Simulations lie behind the many types of skills microbiome statisticians must master [19; 20]. They help us to...

&lt;img src="figures/noun-labs-99456.png" width=40/&gt; **Design experiments** that have high power to detect subtle signals.
&lt;br/&gt;

&lt;img src="figures/noun-benchmark-7569457.png" width=40/&gt; **Benchmark methods** and identify gaps in the literature.
&lt;br/&gt;

&lt;img src="figures/noun-checkmark-7518321.png" width=40/&gt; **Check conclusions** that might be sensitive to technical processing steps.

It's worthwhile to treat simulation design with the same formality as methods
development.

---

### Semisynthetic Data

One of the major advances has been the design of algorithms that can leverage
public data resources, like [21; 22; 23; 24].

* **Semisynthetic Data**: The output from a simulator that has been designed to mimic external, template data. 
* **Template Data**: Previously gathered experimental data that can be used to train a simulator.

.center[
&lt;img src="figures/template_defn.png" width=670/&gt;
]

---

### Example 1: Microbiome Network Inference

1. Benchmarking methods for microbiome network inference is challenging. We
can't directly observe microbe-microbe interactions, which stands in the way of
ground truth labeling.

1. We use the following choices in the design of our simulation,

- **Template**: American Gut Project (261 samples, 45 most abundant taxa)
- **Estimator**: Zero-Inflated Negative Binomial Copula
- **Goodness-of-Fit**: Graphical Checks
- **Ground Truth**: Copula correlation matrix
- **Summarization**: Estimated vs. ground-truth correlation

---

### Simulation Mechanism

We used scDesign3 [25] to estimate a
Zero-Inflated Negative Binomial Copula with the formula `~ log(sequencing_depth) + BMI` for each taxon. These are example simulated marginals.

.center[
&lt;img src="figures/zinb_marginals.png" width=700/&gt;
]

---

### Estimated Correlation

The resulting correlation estimate lacks substantial banding or blocks. We can
also sanity check some of the highly correlated pairs.

.pull-left[
&lt;img src="figures/estimated_correlation.png" width=900/&gt;
]
.pull-right[
&lt;img src="figures/correlated_pair.png" width=400/&gt;
]

---

### Establishing Ground Truth

To create a basis for methods comparison, we modified the ZINB copula to use a
block correlation matrix with varying intra-block correlation strength.

.center[
  &lt;img src="figures/ground_truth_corr.png" width=450/&gt;
]

---

### Methods Comparison

We compared SpiecEasi , a method designed for microbiome networks,
with the Ledoit-Wolf estimator  on `\(\log\left(1 + x\right)\)` transformed counts.

.center[
  &lt;img src="figures/comparison_difference.png" width=800/&gt;
]

---

### Methods Comparison

We compared SpiecEasi , a method designed for microbiome networks,
with the Ledoit-Wolf estimator  on `\(\log\left(1 + x\right)\)` transformed counts.

.center[
  &lt;img src="figures/correlation_comparison_scatter.png" width=700/&gt;
]

---

### Generalizing

In our review, we consider a wider range of underlying correlation matrices.

.center[
  &lt;img src="figures/network_comparison.png" width=1200/&gt;
]

---

### Example 2: Mediation Analysis

Working with microbiologists and psychologists at UW-Madison, we re-analyzed a
dataset about the gut-brain axis.

.pull-left[
1. We re-analyzed the study [26], which gathered data
from 54 participants assigned to either a mindfulness training intervention or a
waitlist control (n = 27 each).

1. The training lasted 2 months. Data were collected at the start, finish, and 2
month follow-up.
]

.pull-right[
&lt;img src="figures/design.png" width=450/&gt;
]

---

### Mediation Analysis

1. We wondered whether mindfulness intervention might affect behavior, which
in turn influences microbiota composition.
2. To explore this, we applied a form of mediation analysis to the 16S
microbiome and survey data [27; 28].

.center[
&lt;img src="figures/mediation-dag-terms.png" width=550/&gt;
]

---

### Synthetic Null Data

We can alter the simulator so that some pathways are "turned off." Estimates
derived from these data provide a reference null distribution.

.center[
&lt;img src="figures/mindfulness-altered.png" width=780/&gt;&lt;br/&gt;
&lt;span style="font-size: 24px;"&gt;
The middle panel comes from a synthetic null: `\(T \nrightarrow M \to Y\)`.
&lt;/span&gt;
]

---

### Synthetic Null Hypothesis Testing

We rank the effects learned from both the real and synthetic null reference
data. The significance threshold is chosen to control the empirical false
discovery based on synthetic data.

.center[
&lt;img src="figures/synthetic_null_testing.png" width=900/&gt;
]

---

### Software and Resources

All the examples I discussed today can be run from online tutorials we've
written to accompany our papers:

* Simulation for Microbiome Analysis ([go.wisc.edu/wnj5p9](https://go.wisc.edu/wnj5p9))
* Generative Models Examples ([go.wisc.edu/ax73qb](https://go.wisc.edu/ax73qb))

The relevant R packages behind these analyses are:

* `multimedia` - Mediation analysis for microbiome data [29].
* `scDesign3` - An existing simulator for single cell data [30; 31; 25].
* `scDesigner` - Under-development version used in the first tutorial.

---

Simulation turns abstract, conceptual questions into simple empirical ones.

.center[
&lt;img src="figures/simulation_summary.png" width=1000/&gt;
]

---

.center[
### Thank you!
]

* Contact: ksankaran@wisc.edu
* Lab Members: Margaret Thairu, Hanying Jiang, Shuchen Yan, Yuliang Peng, Langtian Ma, Helena Huang
* Funding: NIGMS R01GM152744, NIAID R01AI184095
* Co-authors: Hanying Jiang, Xinran Miao, Mara Beebe, Dan W. Grupe, Richie
Davidson, Jo Handelsman, Saritha Kodikara, Jingyi Jessica Li, Kim-Anh Lê Cao,
Susan Holmes

---

class: reference

### References

[1] V. Gopalakrishnan et al. "The Influence of the Gut Microbiome on Cancer, Immunity, and Cancer Immunotherapy". In: _Cancer Cell_ 33.4 (Apr. 2018), p. 570–580. ISSN: 1535-6108.
DOI: [10.1016/j.ccell.2018.03.015](https://doi.org/10.1016%2Fj.ccell.2018.03.015). URL:
[http://dx.doi.org/10.1016/j.ccell.2018.03.015](http://dx.doi.org/10.1016/j.ccell.2018.03.015).

[2] L. Liu et al. "The Potential of the Gut Microbiome to Reshape the Cancer Therapy Paradigm: A Review". In: _JAMA Oncology_ 8.7 (Jul. 2022), p. 1059. ISSN: 2374-2437. DOI:
[10.1001/jamaoncol.2022.0494](https://doi.org/10.1001%2Fjamaoncol.2022.0494). URL: [http://dx.doi.org/10.1001/jamaoncol.2022.0494](http://dx.doi.org/10.1001/jamaoncol.2022.0494).

[3] B. A. Helmink et al. "The microbiome, cancer, and cancer therapy". In: _Nature Medicine_ 25.3 (Mar. 2019), p. 377–388. ISSN: 1546-170X. DOI:
[10.1038/s41591-019-0377-7](https://doi.org/10.1038%2Fs41591-019-0377-7). URL: [http://dx.doi.org/10.1038/s41591-019-0377-7](http://dx.doi.org/10.1038/s41591-019-0377-7).

[4] N. Cullin et al. "Microbiome and cancer". In: _Cancer Cell_ 39.10 (Oct. 2021), p. 1317–1341. ISSN: 1535-6108. DOI:
[10.1016/j.ccell.2021.08.006](https://doi.org/10.1016%2Fj.ccell.2021.08.006). URL: [http://dx.doi.org/10.1016/j.ccell.2021.08.006](http://dx.doi.org/10.1016/j.ccell.2021.08.006).

[5] M. Castellarin et al. "Fusobacterium nucleatum infection is prevalent in human colorectal carcinoma". In: _Genome Research_ 22.2 (Oct. 2011), p. 299–306. ISSN: 1088-9051. DOI:
[10.1101/gr.126516.111](https://doi.org/10.1101%2Fgr.126516.111). URL: [http://dx.doi.org/10.1101/gr.126516.111](http://dx.doi.org/10.1101/gr.126516.111).

[6] C. Ma et al. "Gut microbiome–mediated bile acid metabolism regulates liver cancer via NKT cells". In: _Science_ 360.6391 (May. 2018). ISSN: 1095-9203. DOI:
[10.1126/science.aan5931](https://doi.org/10.1126%2Fscience.aan5931). URL: [http://dx.doi.org/10.1126/science.aan5931](http://dx.doi.org/10.1126/science.aan5931).

[7] N. Hartmann et al. "Cancer immunity thwarted by the microbiome". In: _Science_ 360.6391 (May. 2018), p. 858–859. ISSN: 1095-9203. DOI:
[10.1126/science.aat8289](https://doi.org/10.1126%2Fscience.aat8289). URL: [http://dx.doi.org/10.1126/science.aat8289](http://dx.doi.org/10.1126/science.aat8289).

[8] J. C. Arthur et al. "Intestinal Inflammation Targets Cancer-Inducing Activity of the Microbiota". In: _Science_ 338.6103 (Oct. 2012), p. 120–123. ISSN: 1095-9203. DOI:
[10.1126/science.1224820](https://doi.org/10.1126%2Fscience.1224820). URL: [http://dx.doi.org/10.1126/science.1224820](http://dx.doi.org/10.1126/science.1224820).

[9] Y. L. Lightfoot et al. "Tailoring gut immune responses with lipoteichoic acid-deficient Lactobacillus acidophilus". In: _Frontiers in Immunology_ 4 (2013). ISSN: 1664-3224. DOI:
[10.3389/fimmu.2013.00025](https://doi.org/10.3389%2Ffimmu.2013.00025). URL: [http://dx.doi.org/10.3389/fimmu.2013.00025](http://dx.doi.org/10.3389/fimmu.2013.00025).

[10] J. Chalif et al. "The Role of the Microbiome in Cancer Therapies". In: _Hematology/Oncology Clinics of North America_ 39.2 (Apr. 2025), p. 269–294. ISSN: 0889-8588. DOI:
[10.1016/j.hoc.2024.12.005](https://doi.org/10.1016%2Fj.hoc.2024.12.005). URL: [http://dx.doi.org/10.1016/j.hoc.2024.12.005](http://dx.doi.org/10.1016/j.hoc.2024.12.005).

[11] R. Francescone et al. "Microbiome, Inflammation, and Cancer". In: _The Cancer Journal_ 20.3 (May. 2014), p. 181–189. ISSN: 1528-9117. DOI:
[10.1097/ppo.0000000000000048](https://doi.org/10.1097%2Fppo.0000000000000048). URL: [http://dx.doi.org/10.1097/PPO.0000000000000048](http://dx.doi.org/10.1097/PPO.0000000000000048).

[12] M. Vétizou et al. "Anticancer immunotherapy by CTLA-4 blockade relies on the gut microbiota". En. In: _Science_ 350.6264 (Nov. 2015), pp. 1079-1084.

---

class: reference

### References

[13] A. Sivan et al. "Commensal Bifidobacterium promotes antitumor immunity and facilitates anti-PD-L1 efficacy". In: _Science_ 350.6264 (2015), pp. 1084-1089.

[14] M. O. Din et al. "Synchronized cycles of bacterial lysis for in vivo delivery". In: _Nature_ 536.7614 (Jul. 2016), p. 81–85. ISSN: 1476-4687. DOI:
[10.1038/nature18930](https://doi.org/10.1038%2Fnature18930). URL: [http://dx.doi.org/10.1038/nature18930](http://dx.doi.org/10.1038/nature18930).

[15] G. D. Poore et al. "RETRACTED ARTICLE: Microbiome analyses of blood and tissues suggest cancer diagnostic approach". In: _Nature_ 579.7800 (Mar. 2020), p. 567–574. ISSN:
1476-4687. DOI: [10.1038/s41586-020-2095-1](https://doi.org/10.1038%2Fs41586-020-2095-1). URL:
[http://dx.doi.org/10.1038/s41586-020-2095-1](http://dx.doi.org/10.1038/s41586-020-2095-1).

[16] A. Gihawi et al. "Major data analysis errors invalidate cancer microbiome findings". In: _mBio_ 14.5 (Oct. 2023). Ed. by I. B. Zhulin. ISSN: 2150-7511. DOI:
[10.1128/mbio.01607-23](https://doi.org/10.1128%2Fmbio.01607-23). URL: [http://dx.doi.org/10.1128/mbio.01607-23](http://dx.doi.org/10.1128/mbio.01607-23).

[17] G. D. Sepich-Poore et al. "Reply to: Caution Regarding the Specificities of Pan-Cancer Microbial Structure".  (Feb. 2023). DOI:
[10.1101/2023.02.10.528049](https://doi.org/10.1101%2F2023.02.10.528049). URL: [http://dx.doi.org/10.1101/2023.02.10.528049](http://dx.doi.org/10.1101/2023.02.10.528049).

[18] G. Tonkin-Hill. _GitHub - gtonkinhill/TCGA\_analysis - github.com_. &lt;https://github.com/gtonkinhill/TCGA_analysis&gt;. [Accessed 21-06-2024]. 2023.

[19] K. Sankaran et al. "Generative Models: An Interdisciplinary Perspective". In: _Annual Review of Statistics and Its Application_ 10.1 (Mar. 2023), p. 325–352. ISSN: 2326-831X.
DOI: [10.1146/annurev-statistics-033121-110134](https://doi.org/10.1146%2Fannurev-statistics-033121-110134). URL:
[http://dx.doi.org/10.1146/annurev-statistics-033121-110134](http://dx.doi.org/10.1146/annurev-statistics-033121-110134).

[20] K. Sankaran et al. "Semisynthetic simulation for microbiome data analysis". En. In: _Brief. Bioinform._ 26.1 (Nov. 2024).

[21] E. Pasolli et al. "Accessible, curated metagenomic data through ExperimentHub". In: _Nature Methods_ 14 (2017), pp. 1023-1024. URL:
[https://api.semanticscholar.org/CorpusID:3403081](https://api.semanticscholar.org/CorpusID:3403081).

[22] E. Muller et al. "The gut microbiome-metabolome dataset collection: a curated resource for integrative meta-analysis". In: _npj Biofilms and Microbiomes_ 8.1 (Oct. 2022). ISSN:
2055-5008. DOI: [10.1038/s41522-022-00345-5](https://doi.org/10.1038%2Fs41522-022-00345-5). URL:
[http://dx.doi.org/10.1038/s41522-022-00345-5](http://dx.doi.org/10.1038/s41522-022-00345-5).

[23] Felix G.M. Ernst &lt;felix.gm.ernst@outlook.com&gt; [aut, cre] (&lt;https://orcid.org/0000-0001-5064-0928&gt;), Leo Lahti [aut] (&lt;https://orcid.org/0000-0001-5537-637X&gt;), Sudarshan Shetty
&lt;sudarshanshetty9@gmail.com&gt; [aut] (&lt;https://orcid.org/0000-0001-7280-9915&gt;). _microbiomeDataSets_. 2021.

[24] _Home - National Microbiome Data Collaborative - microbiomedata.org_. &lt;https://microbiomedata.org/&gt;. [Accessed 17-02-2025].

---

class: reference

### References

[25] D. Song et al. "scDesign3 generates realistic in silico data for multimodal single-cell and spatial omics". In: _Nature Biotechnology_ 42.2 (May. 2023), p. 247–252. ISSN:
1546-1696. DOI: [10.1038/s41587-023-01772-1](https://doi.org/10.1038%2Fs41587-023-01772-1). URL:
[http://dx.doi.org/10.1038/s41587-023-01772-1](http://dx.doi.org/10.1038/s41587-023-01772-1).

[26] D. W. Grupe et al. "The Impact of Mindfulness Training on Police Officer Stress, Mental Health, and Salivary Cortisol Levels". In: _Frontiers in Psychology_ 12 (Sep. 2021).
ISSN: 1664-1078. DOI: [10.3389/fpsyg.2021.720753](https://doi.org/10.3389%2Ffpsyg.2021.720753). URL:
[http://dx.doi.org/10.3389/fpsyg.2021.720753](http://dx.doi.org/10.3389/fpsyg.2021.720753).

[27] K. Imai et al. "A general approach to causal mediation analysis." In: _Psychological methods_ 15.4 (2010), p. 309.

[28] M. B. Sohn et al. "Compositional mediation analysis for microbiome studies". In: _The Annals of Applied Statistics_ 13.1 (2019), pp. 661-681.

[29] H. Jiang et al. "Multimedia: multimodal mediation analysis of microbiome data". In: _Microbiology Spectrum_ 13.2 (Feb. 2025). Ed. by J. Claesen. ISSN: 2165-0497. DOI:
[10.1128/spectrum.01131-24](https://doi.org/10.1128%2Fspectrum.01131-24). URL: [http://dx.doi.org/10.1128/spectrum.01131-24](http://dx.doi.org/10.1128/spectrum.01131-24).

[30] W. V. Li et al. "A statistical simulator scDesign for rational scRNA-seq experimental design". In: _Bioinformatics_ 35.14 (Jul. 2019), p. i41–i50. ISSN: 1367-4811. DOI:
[10.1093/bioinformatics/btz321](https://doi.org/10.1093%2Fbioinformatics%2Fbtz321). URL:
[http://dx.doi.org/10.1093/bioinformatics/btz321](http://dx.doi.org/10.1093/bioinformatics/btz321).

[31] T. Sun et al. "scDesign2: a transparent simulator that generates high-fidelity single-cell gene expression count data with gene correlations captured". In: _Genome Biology_ 22.1
(May. 2021). ISSN: 1474-760X. DOI: [10.1186/s13059-021-02367-2](https://doi.org/10.1186%2Fs13059-021-02367-2). URL:
[http://dx.doi.org/10.1186/s13059-021-02367-2](http://dx.doi.org/10.1186/s13059-021-02367-2).

---

### Evaluation Taxonomy

This is how some common techniques fall into this taxonomy.

* **Graphical, Narrow**: Boxplots or cumulative distribution function plots comparing real vs. simulated taxa (like the DA example).

* **Graphical, Broad**: Principal component plots of real vs. simulated dataset.

* **Quantitative, Narrow**: Two-sample Kolmogorov-Smirnov test.

* **Quantitative, Broad**: Evaluation through classification (next example).

* **Fit-for-Purpose**: Linear model coefficients on real vs. simulated data ([20], "Batch Effect Correction").

---

### SPLS-DA Intuition

We "blend" columns of `\(\mathbf{X}\)` and `\(\mathbf{Y}\)` within tables until the patterns look similar.

.center[
&lt;img src="figures/PLS-step1.png" width=500/&gt;
]

Roughly, choose weights `\(\mathbf{a}\)` and `\(\mathbf{b}\)` to maximize
`\(\text{cor}\left(\mathbf{Xa}, \mathbf{Yb}\right)\)`.

---

### SPLS-DA Intuition

We "blend" columns of `\(\mathbf{X}\)` and `\(\mathbf{Y}\)` within tables until the patterns look similar.

.center[
&lt;img src="figures/PLS-step2.png" width=240/&gt;
]

Roughly, choose weights `\(\mathbf{a}\)` and `\(\mathbf{b}\)` to maximize
`\(\text{cor}\left(\mathbf{Xa}, \mathbf{Yb}\right)\)`.


---

### SPLS-DA Intuition

We "blend" columns of `\(\mathbf{X}\)` and `\(\mathbf{Y}\)` within tables until the patterns look similar.

.center[
&lt;img src="figures/PLS-step3.png" width=400/&gt;
]

Roughly, choose weights `\(\mathbf{a}\)` and `\(\mathbf{b}\)` to maximize
`\(\text{cor}\left(\mathbf{Xa}, \mathbf{Yb}\right)\)`.

---

### SPLS-DA Intuition

We "blend" columns of `\(\mathbf{X}\)` and `\(\mathbf{Y}\)` within tables until the patterns look similar.

.center[
&lt;img src="figures/PLS-step4.png" width=150/&gt;
]

Roughly, choose weights `\(\mathbf{a}\)` and `\(\mathbf{b}\)` to maximize
`\(\text{cor}\left(\mathbf{Xa}, \mathbf{Yb}\right)\)`.

---

### SPLS-DA Intuition

Now we can compare samples from the two tables in a single, shared space.

.center[
&lt;img src="figures/PLS-step5.png" width=800/&gt;
]

---

### SPLS-DA Intuition

Now we can compare samples from the two tables in a single, shared space.

.center[
&lt;img src="figures/PLS-step6.png" width=800/&gt;
]

---

### SPLS-DA Intuition

To get more than one dimension, we can repeat this process after removing any
correlation with previously found patterns.

.center[
&lt;img src="figures/PLS-step7.png" width=800/&gt;
]

---

### Copula Models

More formally, let `\(F_{1}, \dots, F_{D}\)` be the target margins and let `\(\Phi\)` be
the CDF of the Gaussian distribution. Gaussian Copula modeling has these steps.

Estimate:

1. Gaussianize the observed `\(\mathbf{x}_{i}\)` to `\(\mathbf{z}_{i} := \left[\Phi^{-1}\left(F_{1}\left(x_{i1}\right)\right), \dots, \Phi^{-1}\left(F_{D}\left(x_{iD}\right)\right)\right]\)`
1. Estimate the covariance `\(\hat{\Sigma}\)` associated with `\(z_{i}\)`

Simulate:

1. Draw `\(\mathbf{z}^\ast \sim \mathcal{N}\left(0, \Sigma\right)\)` 
1. Transform back `\(\mathbf{x}^{\ast} := \left[F_{1}^{-1}\left(\Phi\left(z_{i1}^\ast\right)\right), \dots, F_{D}^{-1}\left(\Phi\left(z_{iD}^\ast\right)\right)\right]\)`

---

### Real vs. Simulated Correlation

.center[
&lt;img src="figures/correlation_histogram.png" width=700/&gt;
]

A detailed explanation is given [here](https://krisrs1128.github.io/microbiome-simulation/multivariate-power-analysis.html#evaluation).

---

### Tuning High-Dimensional Covariance Estimator

.center[
&lt;img src="figures/covariance_hyperparameter_errors.png" width=700/&gt;
]

A detailed explanation is given [here](https://krisrs1128.github.io/microbiome-simulation/multivariate-power-analysis.html#evaluation).

---

### Intuition

* In the Gaussianized space, it's easy to model correlation.
* The mapping back and forth is possible because we know the margins `\(F\)`.
  - `\(\Phi\)` represents the Gaussian CDF applied componentwise
&lt;br/&gt;
&lt;br/&gt;

.center[
&lt;img src="figures/copula_transformation.png" width=700/&gt;
]

---

### Pilot Study

.pull-left[
1. We re-analyzed a pilot study from 2021 [26], which
gathered data from 54 participants randomly assigned to either a mindfulness
training intervention or a waitlist control (n = 27 each).

1. The training lasted 2 months. Data were collected at the start, finish, and 2
month follow-up.
]

.pull-right[
&lt;img src="figures/design.png" width=450/&gt;
]

---

### Estimated Indirect Effects

These figures summarize the paths `\(T \to M \to Y\)`.&lt;/br&gt;
(i.e., color `\(\to\)` x-axis `\(\to\)` y-axis).

.center[
&lt;img src="figures/mindfulness-indirect-effects.png" width=900/&gt;
]

---

### Figure Sources

frustration by Rikas Dzihab from &lt;a href="https://thenounproject.com/browse/icons/term/frustration/" target="_blank" title="frustration Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

confused by Rikas Dzihab from &lt;a href="https://thenounproject.com/browse/icons/term/confused/" target="_blank" title="confused Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

Benchmark by Sofiah from &lt;a href="https://thenounproject.com/browse/icons/term/benchmark/" target="_blank" title="Benchmark Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

checkmark by Asa Kharisma Dini from &lt;a href="https://thenounproject.com/browse/icons/term/checkmark/" target="_blank" title="checkmark Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)

Lab glassware by Vectors Market from &lt;a href="https://thenounproject.com/browse/icons/term/lab-glassware/" target="_blank" title="Lab glassware Icons"&gt;Noun Project&lt;/a&gt; (CC BY 3.0)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
  "highlightStyle": "github",
  "highlightLines": true,
  "ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
